--- old/src/Data/Acid/TemplateHaskell.hs	2014-03-05 10:04:23.000000000 -0800
+++ new/src/Data/Acid/TemplateHaskell.hs	2014-03-05 10:57:11.415416428 -0800
@@ -1,4 +1,4 @@
-{-# LANGUAGE TemplateHaskell #-}
+{-# LANGUAGE CPP, TemplateHaskell #-}
 {- Holy crap this code is messy. -}
 module Data.Acid.TemplateHaskell
     ( makeAcidic
@@ -257,8 +257,13 @@
              structType = foldl appT (conT eventStructName) [ varT tyvar | PlainTV tyvar <- tyvars ]
          instanceD (cxt $ [ classP classPred [varT tyvar] | PlainTV tyvar <- tyvars, classPred <- preds ] ++ map return context)
                    (return ty)
+#if MIN_VERSION_template_haskell(2,9,0)
+                   [ tySynInstD ''MethodResult (tySynEqn [structType] (return resultType))
+                   , tySynInstD ''MethodState  (tySynEqn [structType] (return stateType))
+#else
                    [ tySynInstD ''MethodResult [structType] (return resultType)
                    , tySynInstD ''MethodState  [structType] (return stateType)
+#endif
                    ]
     where (tyvars, context, _args, stateType, resultType, _isUpdate) = analyseType eventName eventType
           eventStructName = mkName (structName (nameBase eventName))
