diff -ru old/debian/Debianize.hs new/debian/Debianize.hs
--- alex.old/debian/Debianize.hs	2015-09-27 10:10:36.482412648 -0700
+++ alex/debian/Debianize.hs	2015-09-27 10:09:29.102414591 -0700
@@ -0,0 +1,31 @@
+import Control.Lens ((.=), (%=))
+import Data.Map as Map
+import Data.Set as Set
+import Debian.AutoBuilder.Details.Versions (seereasonDefaults)
+import Debian.Debianize hiding (backups)
+import Debian.Relation (BinPkgName(..), Relation(Rel), SrcPkgName(..))
+
+main :: IO ()
+main = performDebianization $ do
+         debianDefaults
+         (debInfo . allowDebianSelfBuildDeps) .= True
+         (debInfo . control . buildDepends) %= (++ [[Rel (BinPkgName "alex") Nothing Nothing]])
+         (debInfo . control . buildDepends) %= (++ [[Rel (BinPkgName "happy") Nothing Nothing]])
+         doExecutable (BinPkgName "alex") (InstallFile {execName = "alex", sourceDir = Nothing, destDir = Nothing, destName = "alex"})
+         (debInfo . compat) .= (Just 9)
+         mapM_ (\ name -> (debInfo . atomSet) %= (Set.insert $ InstallData (BinPkgName "alex") name name))
+                   [ "AlexTemplate"
+                   , "AlexTemplate-debug"
+                   , "AlexTemplate-ghc"
+                   , "AlexTemplate-ghc-debug"
+                   , "AlexTemplate-ghc-nopred"
+                   , "AlexWrapper-basic"
+                   , "AlexWrapper-basic-bytestring"
+                   , "AlexWrapper-gscan"
+                   , "AlexWrapper-monad"
+                   , "AlexWrapper-monad-bytestring"
+                   , "AlexWrapper-monadUserState"
+                   , "AlexWrapper-monadUserState-bytestring"
+                   , "AlexWrapper-posn"
+                   , "AlexWrapper-posn-bytestring"
+                   , "AlexWrapper-strict-bytestring"]
