--- old/Distribution/ParseUtils.hs	2014-06-05 13:27:16.967502264 -0700
+++ new/Distribution/ParseUtils.hs	2014-06-06 06:28:25.469213600 -0700
@@ -267,24 +267,7 @@
    vcat [ ppField name (getter x) | FieldDescr name getter _ <- fields ]
 
 ppField :: String -> Doc -> Doc
-ppField name fielddoc 
-   | isEmpty fielddoc         = empty
-   | name `elem` nestedFields = text name <> colon $+$ nest indentWith fielddoc
-   | otherwise                = text name <> colon <+> fielddoc
-   where
-      nestedFields =
-         [ "description"
-         , "build-depends"
-         , "data-files"
-         , "extra-source-files"
-         , "extra-tmp-files"
-         , "exposed-modules"
-         , "c-sources"
-         , "extra-libraries"
-         , "includes"
-         , "install-includes"
-         , "other-modules"
-         ]
+ppField name fielddoc = text name <> colon <+> fielddoc
 
 showFields :: [FieldDescr a] -> a -> String
 showFields fields = render . ($+$ text "") . ppFields fields
--- old/src/Debian/Orphans.hs	2014-09-18 07:06:01.000000000 -0700
+++ new/src/Debian/Orphans.hs	2014-10-31 16:29:26.036764749 -0700
@@ -44,8 +44,10 @@
 #endif
 
 deriving instance Ord Language
+#if !MIN_VERSION_Cabal(1,21,1)
 deriving instance Ord KnownExtension
 deriving instance Ord Extension
+#endif
 deriving instance Eq Compiler
 deriving instance Ord Compiler
 deriving instance Ord NameAddr
