diff -ruN old/debian/Debianize.hs new/debian/Debianize.hs
--- old/debian/Debianize.hs	1969-12-31 16:00:00.000000000 -0800
+++ new/debian/Debianize.hs	2013-12-18 10:50:53.390180184 -0800
@@ -0,0 +1,11 @@
+import Debian.Debianize ((+=), debianization, DebT, evalDebT, newAtoms, Top(Top), utilsPackageNames, writeDebianization)
+import Debian.Relation (BinPkgName(BinPkgName))
+import Debian.AutoBuilder.Details.Atoms (seereasonDefaultAtoms)
+top :: Top
+top = Top "."
+
+main :: IO ()
+main = newAtoms >>= evalDebT (debianization top seereasonDefaultAtoms customize >> writeDebianization top)
+
+customize :: DebT IO ()
+customize = utilsPackageNames += BinPkgName "cabal-install"
