--- old/cabal-install.cabal	2014-02-19 10:03:21.000000000 -0800
+++ new/cabal-install.cabal	2014-02-19 13:19:18.718665905 -0800
@@ -114,7 +114,7 @@
     -- NOTE: when updating build-depends, don't forget to update version regexps
     -- in bootstrap.sh.
     build-depends:
-        array      >= 0.1      && < 0.5,
+        array      >= 0.1,
         base       >= 4        && < 5,
         bytestring >= 0.9      && < 1,
         Cabal      >= 1.18.0   && < 1.19,
diff -ruN old/debian/Debianize.hs new/debian/Debianize.hs
--- old/debian/Debianize.hs	1969-12-31 16:00:00.000000000 -0800
+++ new/debian/Debianize.hs	2013-12-18 10:50:53.390180184 -0800
@@ -0,0 +1,11 @@
+import Debian.Debianize ((+=), debianization, DebT, evalDebT, newAtoms, seereasonDefaultAtoms, Top(Top), utilsPackageNames, writeDebianization)
+import Debian.Relation (BinPkgName(BinPkgName))
+
+top :: Top
+top = Top "."
+
+main :: IO ()
+main = evalDebT (debianization top seereasonDefaultAtoms customize >> writeDebianization top) newAtoms
+
+customize :: DebT IO ()
+customize = utilsPackageNames += BinPkgName "cabal-install"
