--- old/cabal-install.cabal	2019-01-01 06:26:58.591393189 -0800
+++ new/cabal-install.cabal	2019-01-01 06:51:05.033257353 -0800
@@ -125,7 +125,7 @@
 
 custom-setup
    setup-depends:
-       Cabal     >= 2.2,
+       Cabal,
        base,
        process   >= 1.1.0.1  && < 1.7,
        filepath  >= 1.3      && < 1.5
@@ -304,7 +304,7 @@
         base16-bytestring >= 0.1.1 && < 0.2,
         binary     >= 0.7.3    && < 0.9,
         bytestring >= 0.10.6.0 && < 0.11,
-        Cabal      >= 2.4.1.0  && < 2.5,
+        Cabal,
         containers >= 0.5.6.2  && < 0.7,
         cryptohash-sha256 >= 0.11 && < 0.12,
         deepseq    >= 1.4.1.1  && < 1.5,
