--- old/certificate.cabal	2013-05-24 23:13:24.000000000 -0700
+++ new/certificate.cabal	2013-05-24 23:16:55.677299558 -0700
@@ -31,7 +31,7 @@
                    , mtl
                    , pem >= 0.1 && < 0.2
                    , asn1-data >= 0.7.1 && < 0.8.0
-                   , crypto-pubkey-types >= 0.3 && < 0.4
+                   , crypto-pubkey-types >= 0.3
                    , cryptohash
                    , containers
                    , directory
