diff -ru old/Data/Certificate/X509/Cert.hs new/Data/Certificate/X509/Cert.hs
--- old/Data/Certificate/X509/Cert.hs	2015-11-29 10:53:21.497498607 -0800
+++ new/Data/Certificate/X509/Cert.hs	2015-11-29 10:41:19.489503017 -0800
@@ -1,3 +1,5 @@
+{-# LANGUAGE FlexibleContexts #-}
+
 module Data.Certificate.X509.Cert
         (
         -- * Data Structure
diff -ru haskell-certificate-1.3.9.orig/Data/Certificate/X509/Ext.hs haskell-certificate-1.3.9/Data/Certificate/X509/Ext.hs
--- old/Data/Certificate/X509/Ext.hs	2015-11-29 10:53:47.785498447 -0800
+++ new/Data/Certificate/X509/Ext.hs	2015-11-29 10:41:06.425503097 -0800
@@ -1,3 +1,5 @@
+{-# LANGUAGE FlexibleContexts #-}
+
 -- |
 -- Module      : Data.Certificate.X509.Ext
 -- License     : BSD-style
