--- old/Clckwrks/IrcBot/Plugin.hs	2018-06-13 09:27:11.746188633 -0700
+++ new/Clckwrks/IrcBot/Plugin.hs	2018-08-09 23:05:33.208834690 -0700
@@ -56,7 +56,7 @@
        mTopDir <- clckTopDir <$> getConfig plugins
        let basePath  = maybe "_state"   (\td -> td </> "_state")   mTopDir -- FIXME
            ircLogDir = maybe "_irclogs" (\td -> td </> "_irclogs") mTopDir
-       acid <- openLocalStateFrom (basePath </> "ircBot") (initialIrcBotState emptyIrcConfig)
+       acid <- openLocalStateFrom (\flag path -> Prelude.appendFile (path <> "/locktag") ("Clckwrks.IrcBot.Plugin.ircBotInit " <> show flag <> "\n")) (basePath </> "ircBot") (initialIrcBotState emptyIrcConfig)
        addCleanup plugins Always (createCheckpointAndClose acid)
        reconnect <- botConnect plugins acid ircLogDir
        let ircBotConfig = IrcBotConfig { ircBotLogDirectory = ircLogDir
