--- old/clckwrks-plugin-page.cabal	2014-07-10 16:44:53.793055491 -0700
+++ new/clckwrks-plugin-page.cabal	2014-07-17 15:39:36.992408517 -0700
@@ -39,7 +39,7 @@
   other-modules:
                        Clckwrks.Page.Verbatim
   build-depends:       base                   >= 4.3 && < 4.8,
-                       aeson                  >= 0.6 && < 0.8,
+                       aeson                  >= 0.6 && < 0.9,
                        acid-state             == 0.12.*,
                        attoparsec             >= 0.10 && < 0.13,
                        clckwrks               >= 0.21 && < 0.23,
