--- old/clckwrks-dot-com.cabal	2014-03-08 17:03:09.000000000 -0800
+++ new/clckwrks-dot-com.cabal	2014-03-29 15:18:40.826352844 -0700
@@ -17,7 +17,8 @@
 
 Flag backups
      Description: enable the backups executable (currently disabled by default do to wacky dependencies not on hackage)
-     Default: False
+     Default: True
+     Manual: True
 
 Executable             clckwrks-dot-com-server
   main-is:             Main.hs
