--- old/clckwrks-plugin-media.cabal	2014-05-30 16:18:27.268409980 -0700
+++ new/clckwrks-plugin-media.cabal	2014-06-05 16:03:01.343763269 -0700
@@ -41,7 +41,7 @@
   Build-depends:
     base                    < 5,
     acid-state             >= 0.12 && < 0.13,
-    attoparsec             >= 0.10 && < 0.12,
+    attoparsec             >= 0.10,
     blaze-html             >= 0.5 && < 0.8,
     cereal                 >= 0.4 && < 0.5,
     clckwrks               >= 0.21 && < 0.23,
