--- old/clckwrks-plugin-page.cabal	2013-10-13 13:47:05.000000000 -0700
+++ new/clckwrks-plugin-page.cabal	2013-10-20 19:50:38.200818980 -0700
@@ -62,7 +62,7 @@
                        text                   == 0.11.*,
                        time                   == 1.4.*,
                        template-haskell       >= 2.7 && <= 2.9,
-                       uuid                   == 1.2.*,
+                       uuid                   >= 1.2,
                        utf8-string            == 0.3.*,
                        web-plugins            == 0.2.*,
                        web-routes             == 0.27.*,
