--- old/clckwrks-dot-com.cabal	2013-02-05 09:38:46.000000000 -0800
+++ new/clckwrks-dot-com.cabal	2013-02-05 09:46:18.421903299 -0800
@@ -17,7 +17,7 @@
 
 Flag backups
      Description: enable the backups executable (currently disabled by default do to wacky dependencies not on hackage)
-     Default: False
+     Default: True
 
 Executable             clckwrks-dot-com-server
   main-is:             Main.hs
@@ -27,7 +27,7 @@
                        clckwrks-theme-clckwrks  == 0.2.*,
                        clckwrks-plugin-bugs     == 0.3.*,
                        clckwrks-plugin-media    == 0.3.*,
-                       containers               == 0.4.*,
+                       containers               >= 0.4,
                        happstack-server         >= 7.0 && < 7.2,
                        hsp                      == 0.7.*,
                        mtl                      >= 2.0 && < 2.2,
