--- old/clckwrks-dot-com.cabal	2013-02-05 09:38:46.000000000 -0800
+++ new/clckwrks-dot-com.cabal	2013-02-05 09:46:18.421903299 -0800
@@ -17,7 +17,7 @@
 
 Flag backups
      Description: enable the backups executable (currently disabled by default do to wacky dependencies not on hackage)
-     Default: False
+     Default: True
 
 Executable             clckwrks-dot-com-server
   main-is:             Main.hs
--- old/debian/Debianize.hs	2013-02-21 17:48:04.000000000 -0800
+++ new/debian/Debianize.hs	2013-02-23 10:39:32.117390461 -0800
@@ -13,14 +13,13 @@
 
 main :: IO ()
 main =
-    do let jstreePath = "/usr/share/clckwrks-$CLCKWRKS/jstree"
-           json2Path  = "/usr/share/clckwrks-$CLCKWRKS/json2"
-
-       log <- inputChangeLog "debian"
-       inputCabalization "." defaultAtoms >>= debianization "." . customize jstreePath json2Path log >>= writeDebianization "."
+    do log <- inputChangeLog top
+       debianize top (return . customize . setL changelog (Just log))
+    where
+      top = Top "."
 
-customize :: FilePath -> FilePath -> ChangeLog -> Atoms -> Atoms
-customize jstreePath json2Path log =
+customize :: Atoms -> Atoms
+customize =
     modL control  (\ y -> y {homepage = Just "http://www.clckwrks.com/"}) .
     modL rulesFragments (insert (pack (Prelude.unlines ["build/clckwrks-dot-com-production::", "\techo CLCKWRKS=`ghc-pkg field clckwrks version | sed 's/version: //'` > debian/default"]))) .
     modL installTo (Map.insertWith Set.union (BinPkgName "clckwrks-dot-com-production") (Set.singleton ("debian/default", "/etc/default/clckwrks-dot-com-production"))) .
@@ -28,11 +27,10 @@
     setL sourceFormat (Just Native3) .
     modL missingDependencies (insert (BinPkgName "libghc-clckwrks-theme-clckwrks-doc")) .
     setL revision (Just "") .
-    doWebsite (BinPkgName "clckwrks-dot-com-production") (theSite jstreePath json2Path (BinPkgName "clckwrks-dot-com-production")) .
+    doWebsite (BinPkgName "clckwrks-dot-com-production") (theSite (BinPkgName "clckwrks-dot-com-production")) .
     doBackups (BinPkgName "clckwrks-dot-com-backups") "clckwrks-dot-com-backups" .
     fixRules .
     tight .
-    setL changelog (Just log) .
     setL compat (Just 7)
 
 serverNames = List.map BinPkgName ["clckwrks-dot-com-production" {- , "clckwrks-dot-com-staging", "clckwrks-dot-com-development" -}]
@@ -55,14 +53,14 @@
                           (BinPkgName "libghc-clckwrks-plugin-bugs-dev", BinPkgName "haskell-clckwrks-plugin-bugs-utils"),
                           (BinPkgName "libghc-clckwrks-dev", BinPkgName "haskell-clckwrks-utils")]) deb serverNames
 
-theSite :: FilePath -> FilePath -> BinPkgName -> Site
-theSite jstreePath json2Path deb =
+theSite :: BinPkgName -> Site
+theSite deb =
     Site { domain = hostname'
          , serverAdmin = "logic@seereason.com"
-         , server = theServer jstreePath json2Path deb }
+         , server = theServer deb }
 
-theServer :: FilePath -> FilePath -> BinPkgName -> Server
-theServer jstreePath json2Path deb =
+theServer :: BinPkgName -> Server
+theServer deb =
           Server { hostname =
                        case deb of
                          BinPkgName "clckwrks-dot-com-production" -> hostname'
@@ -78,8 +76,8 @@
                      , "--enable-analytics"
                      , "--jquery-path", "/usr/share/javascript/jquery/"
                      , "--jqueryui-path", "/usr/share/javascript/jquery-ui/"
-                     , "--jstree-path", jstreePath
-                     , "--json2-path",json2Path
+                     , "--jstree-path", "/usr/share/clckwrks-$CLCKWRKS/jstree"
+                     , "--json2-path", "/usr/share/clckwrks-$CLCKWRKS/json2"
                      ]
                  , installFile =
                      InstallFile { execName   = "clckwrks-dot-com-server"
--- old/Backups.hs	2013-02-23 11:52:14.221627383 -0800
+++ new/Backups.hs	2013-02-23 11:50:34.049621938 -0800
@@ -1,5 +1,6 @@
 module Main where
 
 import System.Archive.Site (BackupTarget(..), backup)
+import Names (live, domain)
 
-main = backup (BackupTarget {app = "clckwrks-dot-com-production", user = "upload", host = "seereason.com", keep = 50})
+main = backup (BackupTarget {app = live, user = "upload", host = domain, keep = 50})
--- old/Names.hs	1969-12-31 16:00:00.000000000 -0800
+++ new/Names.hs	2013-02-23 11:51:21.997624540 -0800
@@ -0,0 +1,6 @@
+{-# OPTIONS -fno-warn-missing-signatures #-}
+module Names where
+
+domain = "seereason.com"
+site = "clckwrks-dot-com"
+live = site ++ "-production"
