--- old/clckwrks-plugin-ircbot.cabal	2013-02-19 14:39:13.000000000 -0800
+++ new/clckwrks-plugin-ircbot.cabal	2013-02-19 15:04:19.215697541 -0800
@@ -42,7 +42,7 @@
     base                    < 5,
     acid-state             >= 0.7,
     attoparsec             == 0.10.*,
-    blaze-html             == 0.5.*,
+    blaze-html             >= 0.5,
     clckwrks               >= 0.16 && < 0.17,
     containers             >= 0.4 && < 0.6,
     directory              >= 1.1 && < 1.3,
