--- old/clckwrks.cabal	2012-08-22 10:37:00.000000000 -0700
+++ new/clckwrks.cabal	2012-08-22 10:37:50.000000000 -0700
@@ -15,6 +15,79 @@
 Cabal-version:       >=1.6
 Data-Files:
     static/admin.css
+    jstree/_lib/jquery.hotkeys.js
+    jstree/_lib/jquery.js
+    jstree/_lib/jquery.cookie.js
+    jstree/jquery.jstree.js
+    jstree/_demo/_inc/class._database.php
+    jstree/_demo/_inc/class.tree.php
+    jstree/_demo/_inc/class._database_i.php
+    jstree/_demo/_inc/__mysql_errors.log
+    jstree/_demo/_install.txt
+    jstree/_demo/file.png
+    jstree/_demo/index.html
+    jstree/_demo/_dump.sql
+    jstree/_demo/config.php
+    jstree/_demo/root.png
+    jstree/_demo/server.php
+    jstree/_demo/folder.png
+    jstree/_docs/cookies.html
+    jstree/_docs/unique.html
+    jstree/_docs/_json_data.json
+    jstree/_docs/json_data.html
+    jstree/_docs/_html_data.html
+    jstree/_docs/sort.html
+    jstree/_docs/html_data.html
+    jstree/_docs/themeroller.html
+    jstree/_docs/_drive.png
+    jstree/_docs/xml_data.html
+    jstree/_docs/index.html
+    jstree/_docs/hotkeys.html
+    jstree/_docs/languages.html
+    jstree/_docs/ui.html
+    jstree/_docs/!style.css
+    jstree/_docs/checkbox.html
+    jstree/_docs/_search_data.json
+    jstree/_docs/syntax/page_white_code.png
+    jstree/_docs/syntax/wrapping.png
+    jstree/_docs/syntax/page_white_copy.png
+    jstree/_docs/syntax/!script.js
+    jstree/_docs/syntax/printer.png
+    jstree/_docs/syntax/!style.css
+    jstree/_docs/syntax/magnifier.png
+    jstree/_docs/syntax/clipboard.swf
+    jstree/_docs/syntax/help.png
+    jstree/_docs/_search_result.json
+    jstree/_docs/crrm.html
+    jstree/_docs/types.html
+    jstree/_docs/core.html
+    jstree/_docs/dnd.html
+    jstree/_docs/logo.png
+    jstree/_docs/search.html
+    jstree/_docs/contextmenu.html
+    jstree/_docs/themes.html
+    jstree/_docs/_xml_nest.xml
+    jstree/_docs/_xml_flat.xml
+    jstree/themes/default/d.gif
+    jstree/themes/default/d.png
+    jstree/themes/default/style.css
+    jstree/themes/default/throbber.gif
+    jstree/themes/apple/d.png
+    jstree/themes/apple/style.css
+    jstree/themes/apple/dot_for_ie.gif
+    jstree/themes/apple/bg.jpg
+    jstree/themes/apple/throbber.gif
+    jstree/themes/default-rtl/d.gif
+    jstree/themes/default-rtl/d.png
+    jstree/themes/default-rtl/dots.gif
+    jstree/themes/default-rtl/style.css
+    jstree/themes/default-rtl/throbber.gif
+    jstree/themes/classic/d.gif
+    jstree/themes/classic/d.png
+    jstree/themes/classic/style.css
+    jstree/themes/classic/dot_for_ie.gif
+    jstree/themes/classic/throbber.gif
+    json2/json2.js
 
 Library
   Build-tools:     trhsx
@@ -138,7 +138,7 @@
                    Paths_clckwrks
 
   Build-depends:
-     acid-state                   >= 0.7 && < 0.9,
+     acid-state                   >= 0.7,
      aeson                        >= 0.5 && < 0.7,
      attoparsec                   == 0.10.*,
      base                           < 5,
--- old/Clckwrks/Acid.hs	2013-02-23 15:23:28.000000000 -0800
+++ new/Clckwrks/Acid.hs	2013-06-10 22:52:47.438864759 -0700
@@ -12,7 +12,7 @@
 import Control.Monad.State         (modify)
 import Data.Acid                   (AcidState, Query, Update, makeAcidic)
 import Data.Acid.Local             (openLocalStateFrom, createArchive, createCheckpointAndClose)
-import Data.Acid.Remote            (acidServer)
+import Data.Acid.Remote            (acidServer, skipAuthenticationCheck)
 import Data.Data                   (Data, Typeable)
 import Data.Maybe                  (fromMaybe)
 import Data.SafeCopy               (base, deriveSafeCopy)
@@ -84,7 +84,7 @@
     bracket (openLocalStateFrom (basePath </> "profileData") initialProfileDataState) (createArchiveCheckpointAndClose) $ \profileData ->
     bracket (openLocalStateFrom (basePath </> "core")        initialCoreState)        (createArchiveCheckpointAndClose) $ \core ->
     bracket (openLocalStateFrom (basePath </> "navBar")      initialNavBarState)      (createArchiveCheckpointAndClose) $ \navBar ->
-    bracket (forkIO (tryRemoveFile (basePath </> "profileData_socket") >> acidServer profileData (UnixSocket $ basePath </> "profileData_socket")))
+    bracket (forkIO (tryRemoveFile (basePath </> "profileData_socket") >> acidServer skipAuthenticationCheck (UnixSocket $ basePath </> "profileData_socket") profileData))
             (\tid -> killThread tid >> tryRemoveFile (basePath </> "profileData_socket"))
             (const $ f (Acid auth profile profileData core navBar))
     where
