--- old/clckwrks-dot-com.cabal	2012-12-06 08:47:42.000000000 -0800
+++ new/clckwrks-dot-com.cabal	2012-12-06 08:49:46.069545349 -0800
@@ -12,7 +12,7 @@
 
 Flag backups
      Description: enable the backups executable (currently disabled by default do to wacky dependencies not on hackage)
-     Default: False
+     Default: True
 
 Executable             clckwrks-dot-com-server
   main-is:             Main.hs
