--- old/crypto-numbers.cabal	2013-08-31 07:11:36.000000000 -0700
+++ new/crypto-numbers.cabal	2013-08-31 07:16:34.703361841 -0700
@@ -17,7 +17,7 @@
   Build-Depends:     base >= 4 && < 5
                    , bytestring
                    , vector
-                   , crypto-random >= 0.0 && < 0.1
+                   , crypto-random >= 0.0
   Exposed-modules:   Crypto.Number.ModArithmetic
                      Crypto.Number.Serialize
                      Crypto.Number.Generate
