--- old/darcs.cabal	2014-08-26 09:54:40.000000000 -0700
+++ new/darcs.cabal	2014-08-26 10:04:17.437008448 -0700
@@ -456,7 +456,7 @@
       cc-options:        -DHAVE_CURL
 
     if flag(http)
-        build-depends:    network >= 2.2 && < 2.5,
+        build-depends:    network-uri,
                           HTTP    >= 4000.0.8 && < 4000.3
         cpp-options:      -DHAVE_HTTP
         x-have-http:
