--- old/darcs.cabal	2014-09-03 16:49:48.186536640 -0700
+++ new/darcs.cabal	2014-09-18 15:56:25.635599524 -0700
@@ -456,7 +456,7 @@
       cc-options:        -DHAVE_CURL
 
     if flag(http)
-        build-depends:    network >= 2.2 && < 2.5,
+        build-depends:    network-uri,
                           HTTP    >= 4000.0.8 && < 4000.3
         cpp-options:      -DHAVE_HTTP
         x-have-http:
