--- old/data-object.cabal	2012-01-20 06:42:12.000000000 -0800
+++ new/data-object.cabal	2012-01-20 10:13:25.147160370 -0800
@@ -20,6 +20,6 @@
                      bytestring >= 0.9.1.4,
                      text >= 0.5,
                      time >= 1.1.4,
-                     failure >= 0.1.0 && < 0.2
+                     failure >= 0.1.0
     exposed-modules: Data.Object
     ghc-options:     -Wall
