--- old/elm-server.cabal	2012-10-26 05:14:32.000000000 -0700
+++ new/elm-server.cabal	2012-10-26 05:15:15.811759457 -0700
@@ -32,7 +32,7 @@
                        parsec >= 3.1.1,
                        blaze-html >= 0.5.1,
                        HTTP >= 4000,
-                       happstack-server == 7.0.2,
+                       happstack-server >= 7.0.2,
                        deepseq,
                        directory,
                        Elm >= 0.5.0
