--- old/elm-server.cabal	2013-06-07 10:46:54.000000000 -0700
+++ new/elm-server.cabal	2013-06-07 12:39:15.687870879 -0700
@@ -32,7 +32,7 @@
                        parsec >= 3.1.1,
                        blaze-html >= 0.5.1,
                        HTTP >= 4000,
-                       happstack-server == 7.1.1 || == 7.1.7 || == 7.0.2,
+                       happstack-server >= 7.0.2,
                        deepseq,
                        filepath,
                        Elm >= 0.8
