--- old/elm-server.cabal	2013-01-08 16:12:40.000000000 -0800
+++ new/elm-server.cabal	2013-01-08 16:17:42.282234177 -0800
@@ -32,7 +32,7 @@
                        parsec >= 3.1.1,
                        blaze-html >= 0.5.1,
                        HTTP >= 4000,
-                       happstack-server == 7.1.1 || == 7.0.2,
+                       happstack-server >= 7.0.2,
                        deepseq,
                        filepath,
                        Elm >= 0.7
