--- old/fay.cabal	2013-10-20 13:31:29.000000000 -0700
+++ new/fay.cabal	2013-10-20 13:35:22.320495680 -0700
@@ -412,7 +412,7 @@
                    , directory
                    , filepath
                    , ghc-paths
-                   , haskell-names >= 0.3.1 && < 0.4
+                   , haskell-names >= 0.3.1
                    , haskell-packages >= 0.2.3.1
                    , haskell-src-exts >= 1.14
                    , language-ecmascript >= 0.15 && < 1.0
@@ -451,7 +451,7 @@
                    , haskell-src-exts >= 1.14
                    , language-ecmascript >= 0.15 && < 1.0
                    , mtl
-                   , optparse-applicative >= 0.6 && < 0.7
+                   , optparse-applicative >= 0.6
                    , process
                    , safe
                    , split
