--- old/src/Facebook/Monad.hs	2014-10-20 09:46:45.018215742 -0700
+++ new/src/Facebook/Monad.hs	2015-01-04 06:15:37.425564587 -0800
@@ -60,14 +60,14 @@
     liftBase = lift . liftBase
 
 instance MonadTransControl (FacebookT auth) where
-    newtype StT (FacebookT auth) a = FbStT { unFbStT :: StT (ReaderT FbData) a }
-    liftWith f = F $ liftWith (\run -> f (liftM FbStT . run . unF))
-    restoreT   = F . restoreT . liftM unFbStT
+    type StT (FacebookT auth) a = StT (ReaderT FbData) a
+    liftWith f = F $ liftWith (\run -> f (run . unF))
+    restoreT   = F . restoreT
 
 instance MonadBaseControl b m => MonadBaseControl b (FacebookT auth m) where
-    newtype StM (FacebookT auth m) a = StMT {unStMT :: ComposeSt (FacebookT auth) m a}
-    liftBaseWith = defaultLiftBaseWith StMT
-    restoreM     = defaultRestoreM   unStMT
+    type StM (FacebookT auth m) a = ComposeSt (FacebookT auth) m a
+    liftBaseWith = defaultLiftBaseWith
+    restoreM     = defaultRestoreM
 
 -- | Since @fb-0.14.8@.
 instance MonadLogger m => MonadLogger (FacebookT auth m) where
