--- old/fb.cabal	2013-02-19 11:37:49.000000000 -0800
+++ new/fb.cabal	2013-02-19 13:00:32.876094050 -0800
@@ -68,12 +68,12 @@
     , transformers-base
     , monad-control
     , resourcet
-    , conduit            >= 0.5     && < 0.6
+    , conduit            >= 0.5
     , data-default
     , http-types
     , http-conduit       >= 1.5     && < 1.9
     , attoparsec         >= 0.10    && < 0.11
-    , attoparsec-conduit >= 0.5     && < 0.6
+    , attoparsec-conduit >= 0.5
     , unordered-containers
     , aeson              >= 0.5     && < 0.7
     , base16-bytestring  >= 0.1
@@ -81,7 +81,7 @@
     , time               >= 1.2     && < 1.5
     , old-locale
     , cereal             == 0.3.*
-    , crypto-api         == 0.10.*
+    , crypto-api         >= 0.10
     , cryptohash         >= 0.7
   extensions:
     DeriveDataTypeable
