--- old/feed.cabal	2018-07-05 11:39:40.412328123 -0700
+++ new/feed.cabal	2018-08-10 09:20:04.070236120 -0700
@@ -62,7 +62,7 @@
     Text.RSS1.Syntax
     Text.RSS1.Utils
   build-depends:
-      base >= 4 && < 4.11
+      base >= 4 && < 4.12
     , old-locale == 1.0.*
     , old-time >= 1 && < 1.2
     , time < 1.9
@@ -87,7 +87,7 @@
     Text.RSS.Tests
     Text.RSS.Utils
   build-depends:
-      base >= 4 && < 4.11
+      base >= 4 && < 4.12
     , HUnit >= 1.2 && < 1.7
     , feed
     , old-locale == 1.0.*
