--- gd/gd.cabal.orig	2011-06-25 12:27:26.000000000 -0700
+++ gd/gd.cabal	2011-09-10 14:29:48.514415016 -0700
@@ -21,7 +21,7 @@
   Extensions: ForeignFunctionInterface
   Exposed-Modules: Graphics.GD, Graphics.GD.ByteString, Graphics.GD.ByteString.Lazy
   Ghc-options: -Wall
-  Extra-libraries: gd, png, z, jpeg, m, fontconfig, freetype, expat
+  Extra-libraries: gd, png, z, jpeg, fontconfig, expat
   Includes: gd.h
   Include-dirs:        cbits
   Install-includes: gd-extras.h
