--- old/src/NLP/GenI/General.hs	2013-06-11 06:33:21.000000000 -0700
+++ new/src/NLP/GenI/General.hs	2013-06-11 06:33:02.000000000 -0700
@@ -412,11 +412,3 @@
     replicate min_ '0'
 showBitVector min_ (BitVector x) =
     showBitVector (min_ - 1) (BitVector (shiftR x 1)) ++ (show $ x .&. 1)
-
--- ----------------------------------------------------------------------
--- JSON
--- ----------------------------------------------------------------------
-
-instance JSON Text where
-    readJSON = fmap T.pack . readJSON
-    showJSON = showJSON . T.unpack
