--- old/GenI.cabal	2013-06-11 05:57:45.000000000 -0700
+++ new/GenI.cabal	2013-06-11 06:22:58.111502942 -0700
@@ -146,7 +146,7 @@
               , json        >= 0.4 && < 0.6
               , mtl         > 1.0
               , process     > 1
-              , split       == 0.1.*
+              , split       >= 0.1
 
 Test-Suite geni-test
  Type:          exitcode-stdio-1.0
--- old/src/NLP/GenI/General.hs	2013-06-11 06:33:21.000000000 -0700
+++ new/src/NLP/GenI/General.hs	2013-06-11 06:33:02.000000000 -0700
@@ -412,11 +412,3 @@
     replicate min_ '0'
 showBitVector min_ (BitVector x) =
     showBitVector (min_ - 1) (BitVector (shiftR x 1)) ++ (show $ x .&. 1)
-
--- ----------------------------------------------------------------------
--- JSON
--- ----------------------------------------------------------------------
-
-instance JSON Text where
-    readJSON = fmap T.pack . readJSON
-    showJSON = showJSON . T.unpack
