--- old/GenI.cabal	2013-11-04 06:36:04.000000000 -0800
+++ new/GenI.cabal	2014-05-20 06:16:30.662553974 -0700
@@ -67,7 +67,7 @@
                , errors
                , filepath    == 1.*
                , hslogger
-               , json >= 0.4.3 && < 0.6
+               , json >= 0.4.3
                , mtl > 1.0
                , old-locale
                , ordered == 0.1.*
@@ -142,7 +142,7 @@
               , filepath    == 1.*
               , yaml-light  == 0.1.*
               , hslogger
-              , json        >= 0.4 && < 0.6
+              , json        >= 0.4
               , mtl         > 1.0
               , process     > 1
 
--- old/src/NLP/GenI/General.hs~	2015-06-10 10:00:02.259689096 -0700
+++ new/src/NLP/GenI/General.hs	2015-06-10 09:59:57.247689240 -0700
@@ -108,10 +108,6 @@
 -- Strings
 -- ----------------------------------------------------------------------
 
-instance Binary Text where
-  put = put . T.encodeUtf8
-  get = liftM T.decodeUtf8 get
-
 isGeniIdentLetter :: Char -> Bool
 isGeniIdentLetter x = isAlphaNum x || x `elem` "_'+-."
 
@@ -412,11 +412,3 @@
     replicate min_ '0'
 showBitVector min_ (BitVector x) =
     showBitVector (min_ - 1) (BitVector (shiftR x 1)) ++ (show $ x .&. 1)
-
--- ----------------------------------------------------------------------
--- JSON
--- ----------------------------------------------------------------------
-
-instance JSON Text where
-    readJSON = fmap T.pack . readJSON
-    showJSON = showJSON . T.unpack
