--- old/ghcjs.cabal	2014-07-01 07:50:03.968213779 -0700
+++ new/ghcjs.cabal	2014-07-01 08:02:47.448235104 -0700
@@ -90,7 +90,7 @@
                      GHCJS
     other-modules: Paths_ghcjs
     build-depends: base           >= 4        && < 5,
-                   Cabal,
+                   Cabal          >= 1.21,
                    ghc            >= 7.8      && < 7.10,
                    directory,
                    filepath,
@@ -172,7 +172,7 @@
                       directory  >= 1   && < 1.3,
                       process    >= 1   && < 1.3,
                       filepath,
-                      Cabal,
+                      Cabal      >= 1.21,
                       binary,
                       bin-package-db,
                       bytestring
@@ -196,7 +196,7 @@
     Build-Depends:    base                 >= 4      && < 5,
                       ghcjs,
                       directory,
-                      Cabal,
+                      Cabal                >= 1.21,
                       mtl,
                       transformers,
                       lens,
