--- old/ghcjs.cabal	2014-07-01 07:50:03.968213779 -0700
+++ new/ghcjs.cabal	2014-07-01 08:02:47.448235104 -0700
@@ -106,7 +106,7 @@
                    aeson          >= 0.7      && < 0.8,
                    text           >= 0.11     && < 1.2,
                    wl-pprint-text >= 1.1      && < 1.2,
-                   lens           >= 4.0      && < 4.2,
+                   lens           >= 4.0      && < 4.3,
                    yaml           >= 0.8      && < 0.9,
                    time,
                    system-filepath,
