--- old/ghcjs.cabal	2014-06-05 15:07:30.495670232 -0700
+++ new/ghcjs.cabal	2014-06-05 15:12:35.259678745 -0700
@@ -75,7 +75,7 @@
                      GHCJS
     other-modules: Paths_ghcjs
     build-depends: base           >= 4        && < 5,
-                   Cabal,
+                   Cabal >= 1.21,
                    ghc            >= 7.8      && < 7.10,
                    directory,
                    filepath,
@@ -90,7 +90,7 @@
                    aeson          >= 0.7      && < 0.8,
                    text           >= 0.11     && < 1.2,
                    wl-pprint-text >= 1.1      && < 1.2,
-                   lens           >= 4.0      && < 4.2,
+                   lens           >= 4.0      && < 4.3,
                    yaml           >= 0.8      && < 0.9,
                    time,
                    system-filepath,
@@ -113,12 +113,13 @@
                    safe                       >= 0.3  && < 0.4,
                    parsec                     >= 3.1  && < 3.2,
                    haskell-src-exts           >= 1.15 && < 1.16,
-                   haskell-src-meta           >= 0.6.0.3  && < 0.7
+                   haskell-src-meta           >= 0.6.0.3  && < 0.7,
+                   filemanip
     exposed: True
     buildable: True
     hs-source-dirs: src
     include-dirs: include
-    GHC-Options: -O2 -fno-warn-orphans -auto-all -Wall
+    GHC-Options: -O2 -fno-warn-orphans -auto-all -Wall -package Cabal-1.21.0.0 -package Cabal-1.18.1.3
 
 Executable ghcjs
     Main-Is:        Main.hs
@@ -126,7 +127,7 @@
     hs-source-dirs: src-bin
     Build-Depends:  base                >= 4    &&  < 5,
                     ghcjs
-    GHC-Options:    -O2 -rtsopts -with-rtsopts=-N -with-rtsopts=-K256m -threaded
+    GHC-Options:    -O2 -rtsopts -with-rtsopts=-N -with-rtsopts=-K256m -threaded -package Cabal-1.21.0.0 -package Cabal-1.18.1.3
 
 Executable haddock-ghcjs
     if os(Windows)
@@ -138,7 +139,7 @@
                     process,
                     filepath,
                     ghcjs
-    GHC-Options:    -O
+    GHC-Options:    -O -package Cabal-1.21.0.0 -package Cabal-1.18.1.3
 
 Executable ghcjs-pkg
     if flag(compiler-only)
@@ -146,12 +147,13 @@
     Main-Is:          Pkg.hs
     Default-Language: Haskell2010
     Hs-Source-Dirs:   src-bin
+    ghc-options:  -package Cabal-1.21.0.0 -package Cabal-1.18.1.3
     Build-Depends:    ghcjs,
                       base       >= 4   && < 5,
                       directory  >= 1   && < 1.3,
                       process    >= 1   && < 1.3,
                       filepath,
-                      Cabal,
+                      Cabal >= 1.21,
                       binary,
                       bin-package-db,
                       bytestring
@@ -169,11 +171,12 @@
     Main-Is:          Boot.hs
     Default-Language: Haskell2010
     Hs-Source-Dirs:   src-bin
+    ghc-options:  -package Cabal-1.21.0.0 -package Cabal-1.18.1.3
     Build-Depends:    base                 >= 4    && < 5,
                       ghcjs,
                       directory,
                       network,
-                      Cabal,
+                      Cabal >= 1.21,
                       ghc,
                       ghc-paths,
                       text                 >= 0.11 && < 1.2,
@@ -191,7 +194,7 @@
     type:             exitcode-stdio-1.0
     Default-Language: Haskell2010
     hs-source-dirs:   test
-    ghc-options: -threaded -O
+    ghc-options: -threaded -O -package Cabal-1.21.0.0 -package Cabal-1.18.1.3
     main-is:          TestRunner.hs
     build-depends:    base                 >= 4 && < 5,
                       process,
