diff -ru ghcjs2/ghcjs.cabal ghcjs/ghcjs.cabal
--- old/ghcjs.cabal	2015-02-09 16:34:24.479995497 -0800
+++ new/ghcjs.cabal	2015-01-05 13:54:15.014115172 -0800
@@ -255,7 +255,7 @@
                       text                 >= 0.11   && < 1.3,
                       bytestring           >= 0.10   && < 0.11,
                       system-filepath      >= 0.4    && < 0.5,
-                      shelly               >= 1.5    && < 1.6,
+                      shelly               >= 1.5    && < 1.7,
                       system-fileio        >= 0.3    && < 0.4,
                       optparse-applicative >= 0.11   && < 0.12,
                       tar                  >= 0.4    && < 0.5,
