--- old/Setup.lhs	2014-06-07 09:32:18.000000000 -0700
+++ new/Setup.lhs	2014-06-12 10:26:32.548347536 -0700
@@ -1,6 +1,15 @@
-#!/usr/bin/runhaskell 
+#!/usr/bin/runhaskell -package Cabal-1.21.0.0
 > module Main where
+> import Distribution.Compiler
 > import Distribution.Simple
+> import Distribution.Simple.Setup
+> import System.Environment
 > main :: IO ()
-> main = defaultMain
-
+> main = do
+>   args <- getArgs
+>   let args' = args ++
+>               case args of
+>                 ["configure"] -> ["--ghcjs", "--package-db=/homedoesnotexistatbuildtime/.ghcjs/x86_64-linux-0.1.0-7.8.2/package.conf.d", "--prefix=/homedoesnotexistatbuildtime"]
+>                 ["install"] -> ["--global"]
+>                 _ -> []
+>   withArgs args' defaultMain
