--- old/Setup.hs	2014-07-08 07:38:57.781088293 -0700
+++ new/Setup.hs	2014-07-08 08:17:32.013152934 -0700
@@ -37,9 +37,12 @@
  -}
 ghcjsSDist :: Args -> SDistFlags -> IO HookedBuildInfo
 ghcjsSDist as flags = do
-  rawSystemExit (fromFlagOrDefault normal $ sDistVerbosity flags) "bash" ["utils/update_archives.sh"]
+  updateArchives (fromFlagOrDefault normal $ sDistVerbosity flags)
   return emptyHookedBuildInfo
 
+updateArchives :: Verbosity -> IO ()
+updateArchives v = rawSystemExit v "bash" ["utils/update_archives.sh"]
+
 ghcjsPostCopy :: Args -> CopyFlags -> PackageDescription -> LocalBuildInfo -> IO ()
 ghcjsPostCopy args flags descr lbi
   | (FlagName "no-wrapper-install", True) `elem` configConfigurationsFlags (configFlags lbi) =

