--- old/happstack-dot-com.cabal	2014-03-29 15:13:14.000000000 -0700
+++ new/happstack-dot-com.cabal	2014-03-29 15:13:41.942360774 -0700
@@ -18,7 +18,8 @@
 
 Flag backups
      Description: enable the backups executable (currently disabled by default do to wacky dependencies not on hackage)
-     Default: False
+     Default: True
+     Manual: True
 
 Executable happstack-dot-com-server
   Main-is: Main.hs
