--- old/happstack-fay.cabal	2013-07-16 12:30:59.403682305 -0700
+++ new/happstack-fay.cabal	2013-07-16 12:29:58.875680875 -0700
@@ -19,8 +19,8 @@
   exposed-modules:     Happstack.Fay
 
   build-depends:       base                 > 4.5 && <5,
-                       fay                  == 0.14.*,
+                       fay                  >= 0.15,
                        happstack-fay-ajax   == 0.2.*,
                        aeson                == 0.6.*,
                        mtl                  >= 2.0 && < 2.2,
-                       happstack-server     >= 7.0 && <7.2
+                       happstack-server     >= 7.0
