--- old/happstack-plugins.cabal	2013-07-16 10:54:35.000000000 -0700
+++ new/happstack-plugins.cabal	2013-07-16 11:42:29.751613559 -0700
@@ -30,8 +30,8 @@
   build-depends:       base             >= 3 && < 5,
                        plugins-auto     == 0.0.*,
                        mtl              >= 2.0 && <2.2,
-                       happstack-server >= 6   && <7.2,
-                       template-haskell >= 2.6 && <2.8,
-                       th-lift          == 0.5.*
+                       happstack-server >= 6,
+                       template-haskell >= 2.6,
+                       th-lift          >= 0.5
 
   ghc-options:      -Wall
