--- old/happstack-authenticate.cabal	2013-02-12 11:24:50.000000000 -0800
+++ new/happstack-authenticate.cabal	2013-02-19 13:33:09.311989592 -0800
@@ -32,7 +32,7 @@
                        acid-state                   >= 0.6,
                        aeson                        >= 0.4 && < 0.7,
                        authenticate                 == 1.3.*,
-                       blaze-html                   == 0.5.*,
+                       blaze-html                   >= 0.5,
                        bytestring                   >= 0.9 && < 0.11,
                        containers                   >= 0.4 && < 0.6,
                        ixset                        >= 1.0 && < 1.1,
