--- old/happstack-util.cabal	2012-12-18 06:49:52.507546733 -0800
+++ new/happstack-util.cabal	2012-12-18 07:05:49.411569340 -0800
@@ -29,7 +29,7 @@
                        directory,
                        extensible-exceptions, 
                        hslogger >= 1.0.2,
-                       mtl >= 1.1 && < 2.1,
+                       mtl >= 1.1,
                        old-locale,
                        old-time,
                        parsec < 4,
