--- old/happstack-fay.cabal	2013-06-08 09:47:10.000000000 -0700
+++ new/happstack-fay.cabal	2013-06-09 06:40:58.107445707 -0700
@@ -19,7 +19,7 @@
   exposed-modules:     Happstack.Fay
 
   build-depends:       base                 > 4.5 && <5,
-                       fay                  == 0.14.*,
+                       fay                  >= 0.15,
                        happstack-fay-ajax   == 0.2.*,
                        aeson                == 0.6.*,
                        mtl                  >= 2.0 && < 2.2,
