--- old/happstack-server-tls.cabal	2018-07-06 07:06:48.939850436 -0700
+++ new/happstack-server-tls.cabal	2018-07-07 12:41:28.956903819 -0700
@@ -32,7 +32,7 @@
                        happstack-server      >= 6.6.4 && < 7.6,
                        hslogger              >=  1.1 && < 1.3,
                        HsOpenSSL             >= 0.10 && < 0.12,
-                       network               >= 2.3 && < 2.7,
+                       network               >= 2.3 && < 2.8,
                        sendfile              == 0.7.*,
                        time                  >= 1.2 && < 1.10
     -- these extra libraries are not needed to build/install the server
