--- old/debian/control
+++ new/debian/control
@@ -43,7 +43,7 @@ Description: Tools to help Debian developers build Haskell packages
 Package: haskell-devscripts
 Architecture: all
 Depends: ${misc:Depends}
-  , ghc (>= 8.2)
+  , ghc (>= 8.0.1)
   , haskell-devscripts-minimal
   , hscolour
 Description: Tools to help Debian developers build Haskell packages
--- old/Dh_Haskell.sh	2016-12-11 16:48:06.000000000 -0800
+++ new/Dh_Haskell.sh	2017-01-23 13:56:39.054461622 -0800
@@ -6,11 +6,11 @@
 }
 
 cpu(){
-  ghc -e 'putStr System.Info.arch'
+  ghc -ignore-dot-ghci -e 'putStr System.Info.arch'
 }
 
 os(){
-  ghc -e 'putStr System.Info.os'
+  ghc -ignore-dot-ghci -e 'putStr System.Info.os'
 }
 
 ghcjs_version(){
@@ -107,7 +107,7 @@
 }
 
 strip_hash(){
-    echo "$1" | sed 's/-................................$//'
+    echo "$1" | sed 's/-[^-]*$//'
 }
 
 sort_uniq(){
@@ -142,19 +142,28 @@
     local dirs
     local lib
     local hc
-    local ghcversion=`dpkg-query --showformat '${Version}' --show ghc`
     hc=$1
-    if dpkg --compare-versions "${ghcversion}" '>=' 8
-    then
-        dep=$2
-    else
-        dep=`strip-hash $2`
-    fi
+    local ghcversion=`dpkg-query --showformat '${Version}' --show ${hc}`
+    case ${hc} in
+	ghcjs)
+          dep=`strip_hash $2`
+	  ext=".js_a"
+	  ;;
+        ghc)
+	  if dpkg --compare-versions "${ghcversion}" '>=' 8
+	  then
+              dep=$2
+	  else
+              dep=`strip_hash $2`
+	  fi
+	  ext=".a"
+	  ;;
+    esac
     dirs=`ghc_pkg_field $hc $dep library-dirs | grep -i ^library-dirs | cut -d':' -f 2`
     lib=`ghc_pkg_field $hc $dep hs-libraries | grep -i ^hs-libraries |  sed -e 's|hs-libraries: *\([^ ]*\).*|\1|' `
     for dir in $dirs ; do
-        if [ -e "${dir}/lib${lib}.a" ] ; then
-            package=`dpkg-query -S ${dir}/lib${lib}.a | cut -d':' -f 1` || exit $?
+        if [ -e "${dir}/lib${lib}${ext}" ] ; then
+            package=`dpkg-query -S ${dir}/lib${lib}${ext} | cut -d':' -f 1` || exit $?
             continue
         fi
     done
@@ -168,13 +177,14 @@
     local dirs
     local lib
     local hc
-    local ghcversion=`dpkg-query --showformat '${Version}' --show ghc`
+    local ghcversion=`dpkg-query --showformat '${version}' --show ghc`
     hc=$1
+    # No prof packages for ghcjs
     if dpkg --compare-versions "${ghcversion}" '>=' 8
     then
         dep=$2
     else
-        dep=`strip-hash $2`
+        dep=`strip_hash $2`
     fi
     dirs=`ghc_pkg_field $hc $dep library-dirs | grep -i ^library-dirs | cut -d':' -f 2`
     lib=`ghc_pkg_field $hc $dep hs-libraries | grep -i ^hs-libraries | sed -e 's|hs-libraries: *\([^ ]*\).*|\1|' `
@@ -231,7 +241,7 @@
     hc=$1
     type=$2
     pkgid=$3
-    ghcpkg="`usable_ghc_pkg`"
+    ghcpkg="`usable_ghc_pkg ${hc}`"
     virtual_pkg=`package_id_to_virtual_package "${hc}" "$type" $pkgid "${ghcpkg}"`
     # As a transition measure, check if dpkg knows about this virtual package
     if dpkg-query -W $virtual_pkg >/dev/null 2>/dev/null;
@@ -295,15 +305,17 @@
 }
 
 usable_ghc_pkg() {
+    local hc
     local ghcpkg
     local version
+    hc=$1
     if [ -x inplace/bin/ghc-pkg ]
     then
         # We are building ghc and need to use the new ghc-pkg
         ghcpkg="inplace/bin/ghc-pkg"
         version="`dpkg-parsechangelog -S Version`"
     else
-        ghcpkg="ghc-pkg"
+        ghcpkg="${hc}-pkg"
         version="`dpkg-query --showformat '${Version}' --show ghc`"
     fi
     # ghc-pkg prior to version 8 is unusable for our purposes.
@@ -314,8 +326,11 @@
 }
 
 tmp_package_db() {
+    local hc
+    hc=$1
+    shift
     local ghcpkg
-    ghcpkg="`usable_ghc_pkg`"
+    ghcpkg="`usable_ghc_pkg ${hc}`"
     if [ -n "${ghcpkg}" ]
     then
         if [ ! -f debian/tmp-db/package.cache ]
@@ -332,8 +347,6 @@
     local hc
     local dep
     local packages
-    hc=$1
-    shift
     ghcpkg="`tmp_package_db $@`"
     for package_id in `cabal_package_ids $@` ; do
         packages="$packages, `package_id_to_virtual_package "${hc}" dev $package_id "${ghcpkg}"`"
@@ -347,7 +360,7 @@
     local packages
     hc=$1
     shift
-    ghcpkg="`tmp_package_db $@`"
+    ghcpkg="`tmp_package_db ${hc} $@`"
     for package_id in `cabal_package_ids $@` ; do
         packages="$packages, `package_id_to_virtual_package "${hc}" prof $package_id "${ghcpkg}"`"
     done
@@ -483,7 +483,7 @@
 check_recipe(){
     # local PS5=$PS4; PS4=" + check_recipe> "; set -x
     hc=`packages_hc`
-    run ${DEB_SETUP_BIN_NAME} test --builddir=dist-${hc} --show-details=direct
+    run ${DEB_SETUP_BIN_NAME} test --builddir=dist-${hc} --show-details=always
     # PS4=$PS5
 }
 
@@ -516,9 +516,9 @@ haddock_recipe(){
 extra_depends_recipe(){
     # local PS5=$PS4; PS4=" + extra_depends_recipe> "; set -x
     hc=$1
     pkg_config=$(LC_ALL=C.UTF-8 ${DEB_SETUP_BIN_NAME} register --builddir=dist-${hc} --gen-pkg-config --verbose=verbose+nowrap | sed -r -n '/^Creating package registration file: /s///p')
     run dh_haskell_extra_depends ${hc} $pkg_config
-    rm $pkg_config
+    rm -r $pkg_config
     # PS4=$PS5
 }
 
@@ -533,10 +533,10 @@ install_dev_recipe(){
     ( run cd debian/tmp-inst-${hc} ; run mkdir -p ${libdir} ; run find ${libdir}/ \
         \( ! -name "*_p.a" ! -name "*.p_hi" ! -type d \) \
         -exec install -Dm 644 '{}' ../${PKG}/'{}' ';' )
     pkg_config=$(LC_ALL=C.UTF-8 ${DEB_SETUP_BIN_NAME} register --builddir=dist-${hc} --gen-pkg-config --verbose=verbose+nowrap | sed -r -n '/^Creating package registration file: /s///p')
     if [ "${HASKELL_HIDE_PACKAGES}" ]; then sed -i 's/^exposed: True$/exposed: False/' $pkg_config; fi
     run install -Dm 644 $pkg_config debian/${PKG}/${pkgdir}/$pkg_config
-    run rm -f $pkg_config
+    run rm -rf $pkg_config
     if [ "z${DEB_GHC_EXTRA_PACKAGES}" != "z" ] ; then
        EP_DIR=debian/${PKG}/usr/lib/haskell-packages/extra-packages
        run mkdir -p $EP_DIR
