diff --git a/Dh_Haskell.sh b/Dh_Haskell.sh
--- a/Dh_Haskell.sh
+++ b/Dh_Haskell.sh
@@ -483,7 +483,7 @@
 check_recipe(){
     # local PS5=$PS4; PS4=" + check_recipe> "; set -x
     hc=`packages_hc`
-    run ${DEB_SETUP_BIN_NAME} test --builddir=dist-${hc} --show-details=direct
+    run ${DEB_SETUP_BIN_NAME} test --builddir=dist-${hc} --show-details=always
     # PS4=$PS5
 }
 
--- a/Dh_Haskell.sh
+++ b/Dh_Haskell.sh
@@ -6,11 +6,11 @@ run () {
 }
 
 cpu(){
-  ghc -e 'putStr System.Info.arch'
+  ghc -ignore-dot-ghci -e 'putStr System.Info.arch'
 }
 
 os(){
-  ghc -e 'putStr System.Info.os'
+  ghc -ignore-dot-ghci -e 'putStr System.Info.os'
 }
 
 ghcjs_version(){
-- 
1.9.1

