--- old/heist.cabal	2014-07-17 13:17:11.824169835 -0700
+++ new/heist.cabal	2014-07-17 14:41:34.492311245 -0700
@@ -139,7 +139,7 @@
 
   build-depends:
     MonadCatchIO-transformers  >= 0.2.1   && < 0.4,
-    aeson                      >= 0.6     && < 0.8,
+    aeson                      >= 0.6     && < 0.9,
     attoparsec                 >= 0.10    && < 0.13,
     base                       >= 4       && < 5,
     blaze-builder              >= 0.2     && < 0.4,
