diff -ru old/HJavaScript.cabal new/HJavaScript.cabal
--- old/HJavaScript.cabal	2018-07-06 07:07:03.888008994 -0700
+++ new/HJavaScript.cabal	2018-07-07 11:00:07.046816577 -0700
@@ -17,6 +17,6 @@
 Hs-Source-Dirs:   src
 Exposed-Modules:  Language.HJavaScript.Syntax
 
-Build-Depends:    base <4.9, pretty >= 1.0
+Build-Depends:    base <4.12, pretty >= 1.0
 Build-Type:       Simple
 Category:         Language
diff -ru old/src/Language/HJavaScript/Syntax.hs new/src/Language/HJavaScript/Syntax.hs
--- old/src/Language/HJavaScript/Syntax.hs	2010-03-05 08:31:18.000000000 -0800
+++ new/src/Language/HJavaScript/Syntax.hs	2018-07-07 11:00:58.415443112 -0700
@@ -67,7 +67,8 @@
     renderBlock,
      
   ) where
-   
+
+import Prelude hiding ((<>))
 import Text.PrettyPrint.HughesPJ
 import Text.Printf(printf)
 import Data.Char
@@ -215,7 +216,7 @@
   showsFeature :: a -> ShowS
 
 -- A class can be a feature
-instance IsClass c => IsFeature c where
+instance (IsClass c, Show c) => IsFeature c where
   showsFeature = shows
   
 
@@ -504,7 +505,7 @@
 class Show r => IsDeref r
 
 -- There are two kinds of dereferencing; either from classes or objects.
-instance IsClass c  => IsDeref c
+instance (IsClass c, Show c)  => IsDeref c
 instance IsClass c  => IsDeref (Exp c)
 
 -------------------------------------------------------------------
