--- old/hsx.cabal	2013-09-02 22:53:13.000000000 -0700
+++ new/hsx.cabal	2013-09-03 00:45:28.318746687 -0700
@@ -41,7 +41,7 @@
 Flag base4
 
 Library
-  Build-depends:	mtl, haskell-src-exts >= 1.13 && < 1.14, utf8-string
+  Build-depends:	mtl, haskell-src-exts >= 1.15, utf8-string
   if flag(base4)
     Build-depends:      base >= 4 && < 5
     cpp-options:        -DBASE4
--- old/src/HSX/Transform.hs	2013-09-02 21:38:11.000000000 -0700
+++ new/src/HSX/Transform.hs	2013-09-02 22:20:18.438621624 -0700
@@ -348,7 +348,7 @@
     App e1 e2         -> transform2exp e1 e2 App
     NegApp e          -> fmap NegApp $ transformExpM e
     If e1 e2 e3       -> transform3exp e1 e2 e3 If
-    Tuple es          -> fmap Tuple $ mapM transformExpM es
+    Tuple bx es       -> fmap (Tuple bx) $ mapM transformExpM es
     List es           -> fmap List $ mapM transformExpM es
     Paren e           -> fmap Paren $ transformExpM e
     LeftSection e op  -> do e' <- transformExpM e
@@ -692,7 +692,7 @@
                                 (\p1 p2 -> PInfixApp p1 n p2)
                                 renameRP
     PApp n ps         -> renameNpat ps (PApp n) renameRP
-    PTuple ps         -> renameNpat ps PTuple renameRP
+    PTuple bx ps      -> renameNpat ps (PTuple bx) renameRP
     PList ps          -> renameNpat ps PList renameRP
     PParen p          -> rename1pat p PParen renameRP
     PRec n pfs        -> renameNpat pfs (PRec n) renameRPf
@@ -756,7 +756,7 @@
                                 (\p1 p2 -> PInfixApp p1 n p2)
                                 renameIrrP
     PApp n ps         -> renameNpat ps (PApp n) renameIrrP
-    PTuple ps         -> renameNpat ps PTuple renameIrrP
+    PTuple bx ps      -> renameNpat ps (PTuple bx) renameIrrP
     PList ps          -> renameNpat ps PList renameIrrP
     PParen p          -> rename1pat p PParen renameIrrP
     PRec n pfs        -> renameNpat pfs (PRec n) renameIrrPf
@@ -984,7 +984,7 @@
     PNeg q             -> tr1pat q PNeg (trPattern s)
     PInfixApp p1 op p2 -> tr2pat p1 p2 (\p1 p2 -> PInfixApp p1 op p2) (trPattern s)
     PApp n ps          -> trNpat ps (PApp n) (trPattern s)
-    PTuple ps          -> trNpat ps PTuple (trPattern s)
+    PTuple bx ps       -> trNpat ps (PTuple bx) (trPattern s)
     PList ps           -> trNpat ps PList (trPattern s)
     PParen p           -> tr1pat p PParen (trPattern s)
     PRec n pfs         -> trNpat pfs (PRec n) (trPatternField s)
@@ -1481,7 +1481,7 @@
             PInfixApp p1 _ p2  -> gatherPVars p1 ++
                                          gatherPVars p2
             PApp _ ps          -> concatMap gatherPVars ps
-            PTuple ps          -> concatMap gatherPVars ps
+            PTuple _ ps        -> concatMap gatherPVars ps
             PList ps           -> concatMap gatherPVars ps
             PParen p           -> gatherPVars p
             PRec _ pfs         -> concatMap help pfs
--- old/src/Trhsx.hs	2013-09-02 22:31:15.000000000 -0700
+++ new/src/Trhsx.hs	2013-09-02 22:36:13.054635331 -0700
@@ -64,7 +64,7 @@
                  transform $ checkParse $ parse fp fc
 
 parse :: String -> String -> ParseResult Module
-parse fn fc = parseModuleWithMode (ParseMode fn allExtensions False True (Just baseFixities)) fcuc
+parse fn fc = parseModuleWithMode (ParseMode fn Haskell2010 (map EnableExtension allExtensions) False True (Just baseFixities)) fcuc
   where fcuc= unlines $ filter (not . isPrefixOf "#") $ lines fc
 
 usageString :: String
--- old/src/HSX/Transform.hs	2014-06-04 18:29:11.193594923 -0700
+++ new/src/HSX/Transform.hs	2014-06-04 18:39:37.401612414 -0700
@@ -992,7 +992,6 @@
     PWildCard          -> return p
     PIrrPat p          -> tr1pat p PIrrPat (trPattern s)
     PatTypeSig s p t   -> tr1pat p (\p -> PatTypeSig s p t) (trPattern s)
-    PExplTypeArg _ _   -> return p
     PQuasiQuote _ _    -> return p
     PBangPat p         -> tr1pat p PBangPat (trPattern s)
     PNPlusK _ _        -> return p
