--- old/hsx2hs.cabal	2014-07-21 16:37:51.546838703 -0700
+++ new/hsx2hs.cabal	2014-10-02 06:19:43.900008518 -0700
@@ -42,7 +42,7 @@
 
 Library
   Build-depends:	mtl              >= 2.0  && < 2.3,
-                        haskell-src-exts >= 1.14 && < 1.16,
+                        haskell-src-exts >= 1.14 && < 1.17,
                         haskell-src-meta >= 0.6  && < 0.7,
                         template-haskell >= 2.7  && < 2.10,
                         utf8-string      == 0.3.*
