--- old/idris.cabal	2014-10-31 05:04:00.000000000 -0700
+++ new/idris.cabal	2014-11-02 05:59:01.450754522 -0800
@@ -701,7 +701,7 @@
                 , filepath < 1.4
                 , fingertree >= 0.1 && < 0.2
                 , haskeline >= 0.7 && < 0.8
-                , lens >= 4.1.1 && < 4.5
+                , lens >= 4.1.1 && < 4.6
                 , mtl < 2.3
                 , network < 2.7
                 , optparse-applicative >= 0.11 && < 0.12
--- old/idris.cabal	2014-05-28 20:55:26.000000000 -0700
+++ new/idris.cabal	2014-05-29 06:41:28.288610346 -0700
@@ -60,6 +60,26 @@
                        rts/idris_stdfgn.h
                        rts/libtest.c
 
+                       libs/prelude/*.ibc
+                       libs/prelude/Decidable/*.ibc
+                       libs/prelude/Prelude/*.ibc
+                       libs/base/*.ibc
+                       libs/base/Control/*.ibc
+                       libs/base/Control/Isomorphism/*.ibc
+                       libs/base/Control/Monad/*.ibc
+                       libs/base/Data/*.ibc
+                       libs/base/Data/Vect/*.ibc
+                       libs/base/Debug/*.ibc
+                       libs/base/Decidable/*.ibc
+                       libs/base/Language/*.ibc
+                       libs/base/Language/Reflection/*.ibc
+                       libs/base/Network/*.ibc
+                       libs/base/System/Concurrency/*.ibc
+                       libs/base/System/*.ibc
+                       libs/base/Syntax/*.ibc
+                       libs/effects/Effect/*.ibc
+                       libs/effects/*.ibc
+
 Extra-source-files:
                        Makefile
                        config.mk
