--- old/io-streams.cabal	2018-07-07 08:13:56.887341798 -0700
+++ new/io-streams.cabal	2018-07-07 08:38:42.580929387 -0700
@@ -123,7 +123,6 @@
   Build-depends:     base               >= 4     && <5,
                      attoparsec         >= 0.10  && <0.14,
                      bytestring         >= 0.9   && <0.11,
-                     bytestring-builder >= 0.10  && <0.11,
                      network            >= 2.3   && <2.8,
                      primitive          >= 0.2   && <0.7,
                      process            >= 1.1   && <1.7,
@@ -206,7 +205,6 @@
   Build-depends:     base               >= 4     && <5,
                      attoparsec         >= 0.10  && <0.14,
                      bytestring         >= 0.9   && <0.11,
-                     bytestring-builder >= 0.10  && <0.11,
                      deepseq            >= 1.2   && <1.5,
                      directory          >= 1.1   && <2,
                      filepath           >= 1.2   && <2,
