--- old/ircbot.cabal	2014-07-21 16:37:53.694838772 -0700
+++ new/ircbot.cabal	2014-09-18 16:36:43.999611213 -0700
@@ -41,11 +41,12 @@
                   filepath   >= 1.2  && < 1.4,
                   irc        == 0.6.*,
                   mtl        >= 2.0  && < 2.3,
-                  network    >= 2.3  && < 2.6,
+                  network    >= 2.6,
+                  network-uri >= 2.6,
                   old-locale == 1.0.*,
                   parsec     == 3.1.*,
                   time       == 1.4.*,
                   unix       >= 2.4  && < 2.8,
-                  random     == 1.0.*,
+                  random     >= 1.0  && < 1.2,
                   stm        >= 2.2  && < 2.5,
                   SafeSemaphore >= 0.10 && < 0.11
