--- old/debian/control	2014-08-26 12:18:32.000000000 -0700
+++ new/debian/control	2014-09-12 11:36:26.391690911 -0700
@@ -3,7 +3,7 @@
 Priority: optional
 Maintainer: Debian Javascript Maintainers <pkg-javascript-devel@lists.alioth.debian.org>
 Uploaders: Marcelo Jorge Vieira (metal) <metal@debian.org>, Steve Kemp <skx@debian.org>
-Build-Depends: debhelper (>= 7.0.50~), node-uglify
+Build-Depends: debhelper (>= 7.0.50~), node-uglify, nodejs-legacy
 Standards-Version: 3.9.4
 Homepage: http://jquery.com/
 Vcs-Browser: http://git.debian.org/?p=pkg-javascript/jquery.git
