--- old/JuicyPixels.cabal	2016-09-26 16:17:51.338017430 -0700
+++ new/JuicyPixels.cabal	2017-01-21 14:38:07.526147754 -0800
@@ -58,7 +58,7 @@
                  binary              >= 0.5     && < 0.9,
                  zlib                >= 0.5.3.1 && < 0.7,
                  transformers        >= 0.2,
-                 vector              >= 0.10    && < 0.12,
+                 vector              >= 0.10    && < 0.13,
                  primitive           >= 0.4     && < 0.7,
                  deepseq             >= 1.1     && < 1.5,
                  containers          >= 0.4.2   && < 0.6
