--- old/lens.cabal	2013-12-09 09:36:53.000000000 -0800
+++ new/lens.cabal	2013-12-09 09:41:14.189010307 -0800
@@ -187,8 +187,7 @@
     tagged                    >= 0.4.4    && < 1,
     template-haskell          >= 2.4      && < 2.10,
     text                      >= 0.11     && < 1.2,
-    transformers              >= 0.2      && < 0.4,
-    transformers-compat       >= 0.1      && < 1,
+    transformers              >= 0.3      && < 0.4,
     unordered-containers      >= 0.2      && < 0.3,
     vector                    >= 0.9      && < 0.11,
     void                      >= 0.5      && < 1
