--- old/lens.cabal	2014-07-17 13:18:03.772171286 -0700
+++ new/lens.cabal	2014-07-17 14:29:47.448291496 -0700
@@ -208,7 +208,7 @@
     transformers              >= 0.2      && < 0.6,
     transformers-compat       >= 0.4      && < 1,
     unordered-containers      >= 0.2.4    && < 0.3,
-    vector                    >= 0.9      && < 0.12,
+    vector                    >= 0.9      && < 0.13,
     void                      >= 0.5      && < 1
 
   if impl(ghc < 8.0)
