--- old/ListLike.cabal	2014-07-19 07:57:49.224461197 -0700
+++ new/ListLike.cabal	2015-01-04 05:20:25.113548578 -0800
@@ -51,7 +51,7 @@
                 ,text       >= 0.11  && < 1.2
                 ,vector     >= 0.5   && < 0.11
                 ,dlist      >= 0.7   && < 0.9
-                ,fmlist     >= 0.8   && < 0.9
+                ,fmlist     >= 0.8   && < 0.10
   
 Test-suite listlike-tests
   Hs-source-dirs: src testsrc
