--- old/logic-TPTP.cabal	2011-09-15 16:31:03.000000000 -0700
+++ new/logic-TPTP.cabal	2011-09-16 13:40:26.458725487 -0700
@@ -53,7 +53,7 @@
 
 
 Library
- ghc-options: -Wall -O2
+ ghc-options: -Wall -O2 -XBangPatterns
 
  build-depends:      base >=4 && < 5
                    , array
--- old/Parser.y	2012-01-20 11:54:31.000000000 -0800
+++ new/Parser.y	2012-01-20 13:07:07.427160933 -0800
@@ -9,7 +9,7 @@
 import Control.Monad
 import Data.List as L
 import Lexer
-import Data.Set as S
+import qualified Data.Set as S
 import Codec.TPTP.Base
 import System.IO
 import System.IO.Unsafe
--- old/ParserC.y	2012-01-20 13:28:19.000000000 -0800
+++ new/ParserC.y	2012-01-20 13:35:49.917165510 -0800
@@ -7,7 +7,7 @@
 import Control.Monad
 import Data.List as L
 import Lexer
-import Data.Set as S
+import qualified Data.Set as S
 import Codec.TPTP.Base
 import System.IO
 import System.IO.Unsafe
