--- old/logic-TPTP.cabal	2011-09-15 16:31:03.000000000 -0700
+++ new/logic-TPTP.cabal	2011-09-16 13:40:26.458725487 -0700
@@ -53,7 +53,7 @@
 
 
 Library
- ghc-options: -Wall -O2
+ ghc-options: -Wall -O2 -XBangPatterns
 
  build-depends:      base >=4 && < 5
                    , array
--- old/Codec/TPTP/Base.hs	2018-07-21 21:01:09.000000000 +0000
+++ new/Codec/TPTP/Base.hs	2019-01-04 04:26:21.124552691 +0000
@@ -17,6 +17,7 @@
 
 import Codec.TPTP.QuickCheck
 import Control.Applicative
+import Control.Monad.Fail (MonadFail(fail))
 import Control.Monad.Identity
 import Control.Monad.State
 import Data.Data
@@ -37,6 +38,9 @@
 import Util
 #endif
 
+instance MonadFail Identity where
+   fail = error
+
 -- * Basic undecorated formulae and terms
 
 -- | Basic (undecorated) first-order formulae
