--- old/mastermind.cabal	2013-03-05 11:35:34.000000000 -0800
+++ new/mastermind.cabal	2013-06-09 09:05:48.687651040 -0700
@@ -31,7 +31,7 @@
                      acid-state         == 0.8.*,
                      aeson              == 0.6.*,
                      containers         >= 0.4 && < 0.6,
-                     fay                == 0.14.*,
+                     fay                >= 0.14,
                      happstack-fay      == 0.2.*,
                      happstack-fay-ajax == 0.2.*,
                      happstack-server   >= 7.0 && < 7.2,
