--- old/openid.cabal	2014-08-25 06:26:39.130685738 -0700
+++ new/openid.cabal	2014-08-25 06:26:21.386686487 -0700
@@ -31,7 +31,7 @@
                    containers >= 0.2.0.0,
                    HTTP       >= 4000.0.9,
                    monadLib   >= 3.6.0.0,
-                   network    >= 2.2.0.0,
+                   network >= 2.6, network-uri >= 2.6,
                    time       >= 1.1.0.0,
                    xml        >= 1.3.0.0,
                    HsOpenSSL  >= 0.9.0.0
@@ -69,7 +69,7 @@
     buildable:     True
     build-depends: base      >= 4.0.0.0 && < 5,
                    monadLib  >= 3.6.0.0,
-                   network   >= 2.2.0.0,
+                   network >= 2.6, network-uri >= 2.6,
                    HsOpenSSL >= 0.9.0.0,
                    openid
 
--- old/src/Data/Digest/OpenSSL/SHA.hs	2014-05-14 10:52:00.000000000 -0700
+++ new/src/Data/Digest/OpenSSL/SHA.hs	2014-05-14 10:59:27.900976851 -0700
@@ -16,6 +16,7 @@
 import Control.Exception
 import Foreign
 import Foreign.C
+import System.IO.Unsafe (unsafePerformIO)
 
 data EVP_MD
 data EVP_MD_CTX
