--- old/openid.cabal	2012-12-16 18:33:41.000000000 -0800
+++ new/openid.cabal	2012-12-16 18:42:05.364473628 -0800
@@ -27,12 +27,13 @@
 
 library
   build-depends:   base       >= 4.0.0.0  && < 5.0.0.0,
-                   bytestring >= 0.9.1.0  && < 0.10.0.0,
-                   containers >= 0.2.0.0  && < 0.4.1.0,
-                   HTTP       >= 4000.0.9 && < 4000.2,
-                   monadLib   >= 3.6.0.0  && < 3.7.0.0,
-                   network    >= 2.2.0.0  && < 2.4.0.0,
-                   time       >= 1.1.0.0  && < 1.3.0.0,
+                   bytestring >= 0.9.1.0,
+                   containers >= 0.2.0.0,
+                   HTTP       >= 4000.0.9,
+                   monadLib   >= 3.6.0.0,
+                   old-exception,
+                   network    >= 2.2.0.0,
+                   time       >= 1.1.0.0,
                    xml        >= 1.3.0.0  && < 1.4.0.0,
                    HsOpenSSL  >= 0.9.0.0  && < 0.11.0.0
   hs-source-dirs:  src
--- old/src/Data/Digest/OpenSSL/AlternativeHMAC.hsc	2012-01-23 15:26:46.027160840 -0800
+++ new/src/Data/Digest/OpenSSL/AlternativeHMAC.hsc	2012-01-20 11:40:25.566842934 -0800
@@ -59,7 +59,7 @@
 showHMAC bs =
     concatMap draw $ BS.unpack bs
     where
-      draw :: (Integral a) => a -> String
+      draw :: (Integral a, Show a) => a -> String
       draw w = case showHex w [] of
                  [x] -> ['0', x]
                  x   -> x
--- old/src/Data/Digest/OpenSSL/SHA.hs	2014-03-07 16:40:44.000000000 -0800
+++ new/src/Data/Digest/OpenSSL/SHA.hs	2014-03-22 06:15:40.801635695 -0700
@@ -14,8 +14,9 @@
 module Data.Digest.OpenSSL.SHA (sha1, sha256) where
 
 import Control.Exception
-import Foreign
+import Foreign hiding (unsafePerformIO)
 import Foreign.C
+import System.IO.Unsafe (unsafePerformIO)
 
 data EVP_MD
 data EVP_MD_CTX
