--- old/OpenGL.hs	2013-10-13 05:17:41.000000000 -0700
+++ new/OpenGL.hs	2013-10-13 05:28:13.667391806 -0700
@@ -55,7 +55,7 @@
 -- | Creates a new OpenGL texture from a bitmap
 makeTextureFromBitmap 
   :: PixelComponent t 
-  => Bitmap t -> Maybe CubeMapTarget -> Level -> PixelFormat -> PixelInternalFormat -> Border -> IO TextureObject
+  => Bitmap t -> Maybe TextureTargetCubeMapFace -> Level -> PixelFormat -> PixelInternalFormat -> Border -> IO TextureObject
 makeTextureFromBitmap bm cubemap level pf pif border = do
   old_binding <- get (textureBinding Texture2D)
   [tex] <- genObjectNames 1 
@@ -67,7 +67,7 @@
 
 texImageFromBitmap
   :: forall t. PixelComponent t 
-  => Bitmap t -> Maybe CubeMapTarget -> Level -> PixelFormat -> PixelInternalFormat -> Border -> IO ()
+  => Bitmap t -> Maybe TextureTargetCubeMapFace -> Level -> PixelFormat -> PixelInternalFormat -> Border -> IO ()
 texImageFromBitmap bm cubemap level pf pif border = do
   withBitmap bm $ \(width,height) nchn pad ptr -> do
 --    old_rowlength <- get (rowLength Unpack)
