--- old/openid.cabal	2012-09-17 16:18:01.000000000 -0700
+++ new/openid.cabal	2012-09-17 21:23:19.021463349 -0700
@@ -28,11 +28,11 @@
 library
   build-depends:   base       >= 4.0.0.0  && < 5.0.0.0,
                    bytestring >= 0.9.1.0  && < 0.10.0.0,
-                   containers >= 0.2.0.0  && < 0.4.1.0,
-                   HTTP       >= 4000.0.9 && < 4000.2,
-                   monadLib   >= 3.6.0.0  && < 3.7.0.0,
-                   network    >= 2.2.0.0  && < 2.4.0.0,
-                   time       >= 1.1.0.0  && < 1.3.0.0,
+                   containers >= 0.2.0.0,
+                   HTTP       >= 4000.0.9,
+                   monadLib   >= 3.6.0.0,
+                   network    >= 2.2.0.0  && < 2.5.0.0,
+                   time       >= 1.1.0.0,
                    xml        >= 1.3.0.0  && < 1.4.0.0,
                    HsOpenSSL  >= 0.9.0.0  && < 0.11.0.0
   hs-source-dirs:  src
--- old/src/Data/Digest/OpenSSL/AlternativeHMAC.hsc	2012-01-23 15:26:46.027160840 -0800
+++ new/src/Data/Digest/OpenSSL/AlternativeHMAC.hsc	2012-01-20 11:40:25.566842934 -0800
@@ -59,7 +59,7 @@
 showHMAC bs =
     concatMap draw $ BS.unpack bs
     where
-      draw :: (Integral a) => a -> String
+      draw :: (Integral a, Show a) => a -> String
       draw w = case showHex w [] of
                  [x] -> ['0', x]
                  x   -> x
