--- old/pandoc.cabal	2019-01-01 20:37:16.050627390 -0800
+++ new/pandoc.cabal	2019-01-01 20:41:09.380012254 -0800
@@ -595,7 +595,10 @@
   other-extensions: NoImplicitPrelude
   hs-source-dirs:  .
   main-is:         pandoc.hs
-  buildable:       True
+  if impl(ghcjs)
+    buildable:       False
+  else
+    buildable:       True
   other-modules:   Paths_pandoc
 
 executable trypandoc
