--- old/pandoc.cabal	2014-09-03 16:47:48.678536062 -0700
+++ new/pandoc.cabal	2014-09-11 10:49:29.439259709 -0700
@@ -226,7 +226,7 @@
                  HTTP >= 4000.0.5 && < 4000.3,
                  texmath >= 0.8 && < 0.9,
                  xml >= 1.3.12 && < 1.4,
-                 random >= 1 && < 1.1,
+                 random >= 1 && < 1.2,
                  extensible-exceptions >= 0.1 && < 0.2,
                  pandoc-types >= 1.12.4 && < 1.13,
                  aeson >= 0.7 && < 0.9,
@@ -253,7 +253,7 @@
   else
      Build-Depends: network >= 2 && < 2.6
   if flag(https)
-     Build-Depends: http-client >= 0.3.2 && < 0.4,
+     Build-Depends: http-client >= 0.3.2 && < 0.5,
                     http-client-tls >= 0.2 && < 0.3,
                     http-types >= 0.8 && < 0.9
      cpp-options:   -DHTTP_CLIENT
