--- old/pandoc.cabal	2014-04-02 15:07:28.000000000 -0700
+++ new/pandoc.cabal	2014-04-02 16:21:14.632343884 -0700
@@ -236,7 +236,7 @@
                  binary >= 0.5 && < 0.8
   Build-Tools:   alex, happy
   if flag(http-conduit)
-     Build-Depends: http-conduit >= 1.9 && < 2.1,
+     Build-Depends: http-conduit >= 1.9,
                     http-types >= 0.8 && < 0.9
      cpp-options:   -DHTTP_CONDUIT
   if flag(embed_data_files)
