--- old/src/Text/Pandoc/Writers/LaTeX.hs
+++ new/src/Text/Pandoc/Writers/LaTeX.hs
@@ -268,13 +268,14 @@ stringToLaTeX  _     []     = return ""
 stringToLaTeX  ctx (x:xs) = do
   opts <- gets stOptions
   rest <- stringToLaTeX ctx xs
+  xelatex <- (elem "xelatex" . writerLaTeXArgs . stOptions) <$> get
   let ligatures = writerTeXLigatures opts && ctx == TextString
   let isUrl = ctx == URLString
   when (x == '€') $
      modify $ \st -> st{ stUsesEuro = True }
   return $
     case x of
-       '€' -> "\\euro{}" ++ rest
+       '€' | not xelatex -> "\\euro{}" ++ rest
        '{' -> "\\{" ++ rest
        '}' -> "\\}" ++ rest
        '`' | ctx == CodeString -> "\\textasciigrave{}" ++ rest
