--- old/pandoc.cabal	2014-09-03 16:47:48.678536062 -0700
+++ new/pandoc.cabal	2014-09-11 10:49:29.439259709 -0700
@@ -253,7 +253,7 @@
   else
      Build-Depends: network >= 2 && < 2.6
   if flag(https)
-     Build-Depends: http-client >= 0.3.2 && < 0.4,
+     Build-Depends: http-client >= 0.3.2 && < 0.5,
                     http-client-tls >= 0.2 && < 0.3,
                     http-types >= 0.8 && < 0.9
      cpp-options:   -DHTTP_CLIENT
