--- old/src/Text/Pandoc/Readers/Markdown.hs	2013-03-20 15:09:15.000000000 -0700
+++ new/src/Text/Pandoc/Readers/Markdown.hs	2013-03-21 07:02:16.778363394 -0700
@@ -1255,7 +1255,7 @@
                 , bareURL
                 , str
                 , endline
-                , code
+                , (guardEnabled Ext_backtick_code_blocks >> code)
                 , fours
                 , strong
                 , emph
