--- old/pandoc.cabal	2016-12-10 02:28:45.000000000 -0800
+++ new/pandoc.cabal	2017-07-18 17:24:48.600181088 -0700
@@ -282,7 +282,7 @@
                  blaze-markup >= 0.5.1 && < 0.8,
                  yaml >= 0.8.8.2 && < 0.9,
                  scientific >= 0.2 && < 0.4,
-                 vector >= 0.10 && < 0.12,
+                 vector >= 0.10 && < 0.13,
                  hslua >= 0.3 && < 0.5,
                  binary >= 0.5 && < 0.9,
                  SHA >= 1.6 && < 1.7,
--- old/src/Text/Pandoc/Writers/LaTeX.hs
+++ new/src/Text/Pandoc/Writers/LaTeX.hs
@@ -268,13 +268,14 @@ stringToLaTeX  _     []     = return ""
 stringToLaTeX  ctx (x:xs) = do
   opts <- gets stOptions
   rest <- stringToLaTeX ctx xs
+  xelatex <- (elem "xelatex" . writerLaTeXArgs . stOptions) <$> get
   let ligatures = writerTeXLigatures opts && ctx == TextString
   let isUrl = ctx == URLString
   when (x == '€') $
      modify $ \st -> st{ stUsesEuro = True }
   return $
     case x of
-       '€' -> "\\euro{}" ++ rest
+       '€' | not xelatex -> "\\euro{}" ++ rest
        '{' -> "\\{" ++ rest
        '}' -> "\\}" ++ rest
        '`' | ctx == CodeString -> "\\textasciigrave{}" ++ rest
