--- old/pandoc.cabal	2013-12-08 09:59:17.000000000 -0800
+++ new/pandoc.cabal	2013-12-08 13:41:07.907033759 -0800
@@ -235,7 +235,7 @@
                  hslua >= 0.3 && < 0.4
   Build-Tools:   alex, happy
   if flag(http-conduit)
-     Build-Depends: http-conduit >= 1.9 && < 1.10,
+     Build-Depends: http-conduit >= 1.9,
                     http-types >= 0.8 && < 0.9
      cpp-options:   -DHTTP_CONDUIT
   if flag(embed_data_files)
