--- old/debian/control	2012-02-08 02:50:28.000000000 -0800
+++ old/debian/control	2013-06-09 21:22:42.908695692 -0700
@@ -18,9 +18,9 @@
 Architecture: any
 Depends: ${haskell:Depends}, ${shlibs:Depends}, ${misc:Depends}
 Suggests: libghc-parsec2-prof, libghc-parsec2-doc
-Provides: libghc-parsec-dev, ${haskell:Provides}
-Conflicts: libghc-parsec-dev (<< 3.0.0-5)
-Replaces: libghc-parsec-dev (<< 3.0.0-5)
+Provides: libghc-parsec-dev, ${haskell:Provides}
+Conflicts: libghc-parsec-dev (<< 3.0.0-5), libghc-parsec3-dev
+Replaces: libghc-parsec-dev (<< 3.0.0-5), libghc-parsec3-dev
 Description: Haskell monadic parser combinator library for GHC
  This package provides a library for the Haskell programming language.
  .
@@ -31,9 +31,9 @@
 Package: libghc-parsec2-prof
 Architecture: any
 Depends: ${haskell:Depends}, ${misc:Depends}
-Provides: libghc-parsec-prof, ${haskell:Provides}
-Conflicts: libghc-parsec-prof (<< 3.0.0-5)
-Replaces: libghc-parsec-prof (<< 3.0.0-5)
+Provides: libghc-parsec-prof, ${haskell:Provides}
+Conflicts: libghc-parsec-prof (<< 3.0.0-5), libghc-parsec3-prof
+Replaces: libghc-parsec-prof (<< 3.0.0-5), libghc-parsec3-prof
 Description: Haskell monadic parser combinator library for GHC; profiling libraries
  This package provides a library for the Haskell programming language,
  compiled for profiling.
@@ -48,9 +48,9 @@
 Depends: ${haskell:Depends}, ${shlibs:Depends}, ${misc:Depends}
 Recommends: ghc-doc, libghc-mtl-doc
 Suggests: libghc-parsec2-dev
-Provides: libghc-parsec-doc
-Conflicts: libghc-parsec-doc (<< 3.0.0-5)
-Replaces: libghc-parsec-doc (<< 3.0.0-5)
+Provides: libghc-parsec-doc
+Conflicts: libghc-parsec-doc (<< 3.0.0-5), libghc-parsec3-doc
+Replaces: libghc-parsec-doc (<< 3.0.0-5), libghc-parsec3-doc
 Description: Haskell monadic parser combinator library for GHC; documentation
  This package provides the documentation for a library for the Haskell
  programming language.
