--- old/src/System/Plugins/Env.hs	2014-09-02 15:01:28.000000000 -0700
+++ new/src/System/Plugins/Env.hs	2014-10-31 08:38:05.704358678 -0700
@@ -69,7 +69,11 @@
 
 import Control.Concurrent.MVar  ( MVar(), newMVar, withMVar )
 
+#if MIN_VERSION_Cabal(1,21,0)
+import Distribution.Package hiding (depends, installedPackageId, packageName, PackageName(..))
+#else
 import Distribution.Package hiding (depends, packageName, PackageName(..))
+#endif
 import Distribution.Text
 
 import Distribution.InstalledPackageInfo
--- old/src/System/Plugins/Load.hs~	2014-07-21 16:38:02.818839067 -0700
+++ new/src/System/Plugins/Load.hs	2015-06-08 12:16:22.948438221 -0700
@@ -72,7 +72,12 @@
 -- import Language.Hi.Parser
 import BinIface
 import HscTypes
-import Module (moduleName, moduleNameString, packageIdString)
+import Module (moduleName, moduleNameString)
+#if __GLASGOW_HASKELL__ < 710
+import Module (packageIdString)
+#else
+import Module (packageKeyString)
+#endif
 import HscMain (newHscEnv)
 import TcRnMonad (initTcRnIf)
 
@@ -701,7 +706,9 @@
 
                 -- and find some packages to load, as well.
                 let ps = dep_pkgs ds
-#if MIN_VERSION_ghc(7,2,0)
+#if MIN_VERSION_ghc(7,10,0)
+                ps' <- filterM loaded . map packageKeyString . nub $ map fst ps
+#elif MIN_VERSION_ghc(7,2,0)
                 ps' <- filterM loaded . map packageIdString . nub $ map fst ps
 #else
                 ps' <- filterM loaded . map packageIdString . nub $ ps
