--- old/reform-happstack.cabal	2014-09-18 15:49:00.563597373 -0700
+++ new/reform-happstack.cabal	2014-09-18 15:49:51.091597618 -0700
@@ -23,7 +23,7 @@
                        bytestring       >= 0.9 && < 0.11,
                        happstack-server >= 7.0 && < 7.4,
                        mtl              >= 2.0 && < 2.3,
-                       random           == 1.0.*,
+                       random           >= 1.0 && < 1.2,
                        reform           == 0.2.*,
                        text             >= 0.11 && < 1.2,
                        utf8-string      == 0.3.*
