--- old/reified-records.cabal	2015-02-20 11:51:05.714937318 -0800
+++ new/reified-records.cabal	2015-02-20 11:51:03.502937382 -0800
@@ -21,7 +21,7 @@
                        Data.Generics.Record.Reify,
                        Data.Generics.Record.Subtype
   other-extensions:    ScopedTypeVariables
-  build-depends:       base ==4.*, containers >=0.5 && <0.6, mtl >=2.1 && <2.2
+  build-depends:       base ==4.*, containers >=0.5 && <0.6, mtl >=2.1 && <2.3
   hs-source-dirs:      src
   default-language:    Haskell2010
   
\ No newline at end of file
