--- old/seereasonpartners-dot-com.cabal	2014-03-29 17:14:10.000000000 -0700
+++ new/seereasonpartners-dot-com.cabal	2014-03-29 17:15:42.790155081 -0700
@@ -18,7 +18,8 @@
 
 Flag backups
      Description: enable the backups executable (currently disabled by default do to wacky dependencies not on hackage)
-     Default: False
+     Default: True
+     Manual: True
 
 Executable seereasonpartners-dot-com-server
   Main-is: Main.hs
