--- old/syb-with-class.cabal	2014-02-18 14:42:19.000000000 -0800
+++ new/syb-with-class.cabal	2014-02-18 17:05:26.200945088 -0800
@@ -25,7 +25,7 @@
   location: http://patch-tag.com/r/Saizan/syb-with-class/
 
 Library 
-    Build-Depends:      base >= 3 && < 5, template-haskell >= 2.4 && < 2.9, bytestring, array, containers
+    Build-Depends:      base >= 3 && < 5, template-haskell >= 2.4, bytestring, array, containers
     Exposed-modules:
         Data.Generics.SYB.WithClass.Basics
         Data.Generics.SYB.WithClass.Context
--- old/Data/Generics/SYB/WithClass/Basics.hs	2014-02-18 21:20:51.000000000 -0800
+++ new/Data/Generics/SYB/WithClass/Basics.hs	2014-02-19 05:05:29.441965852 -0800
@@ -12,13 +12,21 @@
 
 module Data.Generics.SYB.WithClass.Basics (
 
+#if MIN_VERSION_base(4,7,0)
+ module Data.OldTypeable,
+#else
  module Data.Typeable,
+#endif
  module Data.Generics.SYB.WithClass.Context,
  module Data.Generics.SYB.WithClass.Basics
 
 ) where
 
+#if MIN_VERSION_base(4,7,0)
+import Data.OldTypeable
+#else
 import Data.Typeable
+#endif
 import Data.Generics.SYB.WithClass.Context
 
 #ifdef __HADDOCK__
--- old/Data/Generics/SYB/WithClass/Instances.hs.orig	2014-02-19 06:26:16.000000000 -0800
+++ new/Data/Generics/SYB/WithClass/Instances.hs	2014-02-19 09:19:35.258326066 -0800
@@ -364,7 +364,7 @@
 eitherDataType :: DataType
 eitherDataType = mkDataType "Prelude.Either" [leftConstr,rightConstr]
 
-instance (Sat (ctx (Either a b)), Data ctx a, Data ctx b) =>
+instance (Typeable2 Either, Sat (ctx (Either a b)), Data ctx a, Data ctx b) =>
           Data ctx (Either a b) where
   gfoldl _ f z (Left a)   = z Left  `f` a
   gfoldl _ f z (Right a)  = z Right `f` a
@@ -523,7 +523,7 @@
          Data ctx TypeRep where
   toConstr _ _   = error "toConstr"
   gunfold _ _ _  = error "gunfold TypeRep"
-  dataTypeOf _ _ = mkNorepType "Data.Typeable.TypeRep"
+  dataTypeOf _ _ = mkNorepType "Data.OldTypeable.TypeRep"
 
 
 ------------------------------------------------------------------------------
@@ -533,7 +533,7 @@
          Data ctx TyCon where
   toConstr _ _   = error "toConstr"
   gunfold _ _ _  = error "gunfold TyCon"
-  dataTypeOf _ _ = mkNorepType "Data.Typeable.TyCon"
+  dataTypeOf _ _ = mkNorepType "Data.OldTypeable.TyCon"
 
 
 ------------------------------------------------------------------------------
@@ -564,7 +564,8 @@
 ------------------------------------------------------------------------------
 
 
-instance Sat (ctx Handle) =>
+instance (Typeable1 ForeignPtr, Typeable Handle,
+          Sat (ctx Handle)) =>
          Data ctx Handle where
   toConstr _ _   = error "toConstr"
   gunfold _ _ _  = error "gunfold Handle"
@@ -604,7 +605,8 @@
 ------------------------------------------------------------------------------
 
 
-instance (Sat (ctx (ForeignPtr a)), Typeable a) =>
+instance (Typeable1 ForeignPtr,
+          Sat (ctx (ForeignPtr a)), Typeable a) =>
           Data ctx (ForeignPtr a) where
   toConstr _ _   = error "toConstr"
   gunfold _ _ _  = error "gunfold ForeignPtr"
@@ -683,7 +685,8 @@
 mapDataType :: DataType
 mapDataType = mkDataType "Data.Map.Map" [emptyMapConstr,insertMapConstr]
 
-instance (Sat (ctx (M.Map a b)), Data ctx a, Data ctx b, Ord a) =>
+instance (Typeable2 M.Map,
+          Sat (ctx (M.Map a b)), Data ctx a, Data ctx b, Ord a) =>
           Data ctx (M.Map a b) where
   gfoldl _ f z m = case M.minViewWithKey m of
                      Nothing -> z M.empty
@@ -707,7 +710,8 @@
 setDataType :: DataType
 setDataType = mkDataType "Data.Set.Set" [emptySetConstr,insertSetConstr]
 
-instance (Sat (ctx (S.Set a)), Data ctx a, Ord a) =>
+instance (Typeable1 S.Set,
+          Sat (ctx (S.Set a)), Data ctx a, Ord a) =>
           Data ctx (S.Set a ) where
   gfoldl _ f z s = case S.minView s of
                      Nothing -> z S.empty
@@ -723,5 +727,8 @@
 
 ------------------------------------------------------------------------------
 
+instance Typeable ByteString
+instance Typeable L.ByteString
+
 $( deriveData [''ByteString] )
 $( deriveData [''L.ByteString] )
