--- old/quickcheck2/test-framework-quickcheck2.cabal	2016-02-11 07:55:21.911120250 -0800
+++ new/quickcheck2/test-framework-quickcheck2.cabal	2016-10-22 15:49:38.148421797 -0700
@@ -25,7 +25,7 @@
         Exposed-Modules:        Test.Framework.Providers.QuickCheck2
 
         Build-Depends:          test-framework >= 0.7.1,
-                                QuickCheck >= 2.4 && < 2.9,
+                                QuickCheck >= 2.4 && < 2.10,
                                 extensible-exceptions >= 0.1.1 && < 0.2.0
         if flag(base3)
                 Build-Depends:          base >= 3 && < 4, random >= 1
