--- old/test-framework-quickcheck2.cabal	2013-03-12 21:20:08.000000000 -0700
+++ new/test-framework-quickcheck2.cabal	2013-03-13 08:19:06.689846529 -0700
@@ -22,7 +22,7 @@
 Library
         Exposed-Modules:        Test.Framework.Providers.QuickCheck2
 
-        Build-Depends:          test-framework >= 0.7.1, QuickCheck >= 2.4 && < 2.6, extensible-exceptions >= 0.1.1 && < 0.2.0
+        Build-Depends:          test-framework >= 0.7.1, QuickCheck >= 2.4, extensible-exceptions >= 0.1.1 && < 0.2.0
         if flag(base3)
                 Build-Depends:          base >= 3 && < 4, random >= 1
         else
