--- old/test-framework-quickcheck2.cabal	2014-03-22 06:05:49.000000000 -0700
+++ new/test-framework-quickcheck2.cabal	2014-03-22 06:25:50.568243856 -0700
@@ -24,7 +24,7 @@
 Library
         Exposed-Modules:        Test.Framework.Providers.QuickCheck2
 
-        Build-Depends:          test-framework >= 0.7.1, QuickCheck >= 2.4 && < 2.7, extensible-exceptions >= 0.1.1 && < 0.2.0
+        Build-Depends:          test-framework >= 0.7.1, QuickCheck >= 2.4, extensible-exceptions >= 0.1.1 && < 0.2.0
         if flag(base3)
                 Build-Depends:          base >= 3 && < 4, random >= 1
         else
