--- old/text-show.cabal	2018-11-02 05:26:06.638501873 -0700
+++ new/text-show.cabal	2018-11-08 10:55:45.421844996 -0800
@@ -157,7 +157,6 @@
                      , base-compat-batteries >= 0.10   && < 0.11
                      , bifunctors            >= 5.1    && < 6
                      , bytestring            >= 0.9    && < 0.11
-                     , bytestring-builder
                      , containers            >= 0.1    && < 0.7
                      , contravariant         >= 0.5    && < 2
                      , generic-deriving      >= 1.11   && < 2
@@ -342,7 +341,6 @@
                      , base-compat-batteries >= 0.10   && < 0.11
                      , base-orphans          >= 0.6    && < 0.9
                      , bytestring            >= 0.9    && < 0.11
-                     , bytestring-builder
                      , deriving-compat       >= 0.3.4  && < 1
                      , generic-deriving      >= 1.11   && < 2
                      , ghc-prim
