--- old/universe.cabal	2013-10-12 15:10:52.000000000 -0700
+++ new/universe.cabal	2013-10-12 15:11:08.254653400 -0700
@@ -34,6 +34,6 @@
                        containers             >=0.1 && <1  ,
                        keys                   >=0.1 && <4  ,
                        mtl                    >=1.0 && <2.2,
-                       representable-functors >=2.4 && <3.2,
+                       representable-functors >=2.4,
                        transformers           >=0.2 && <0.4,
                        void                   >=0.1 && <0.7
