--- old/urlencoded.cabal	2014-08-24 13:41:37.000000000 -0700
+++ new/urlencoded.cabal	2014-08-24 14:00:46.921182196 -0700
@@ -20,8 +20,16 @@
   Description: Build the test executable
   Default: False
 
+flag network-uri
+   description: Get Network.URI from the network-uri package
+   default: True
+
 Library
-  build-depends:       base == 4.*, network >= 2.2 && < 4, mtl >= 1.1, split == 0.2.*
+  build-depends:       base == 4.*, mtl >= 1.1, split == 0.2.*
+  if flag(network-uri)
+    build-depends: network-uri >= 2.6
+  else
+    build-depends: network-uri >= 2.2 && < 2.6
   ghc-options:         -Wall
   hs-source-dirs:      src
   exposed-modules:     Data.URLEncoded
