--- old/uuid.cabal	2013-10-12 11:15:55.000000000 -0700
+++ new/uuid.cabal	2013-10-12 15:07:05.358649912 -0700
@@ -29,7 +29,7 @@
  Build-Depends: base >=3 && < 5,
                 binary >= 0.4 && < 0.8,
                 bytestring >= 0.9 && < 1.1,
-                cryptohash >= 0.7 && < 0.11,
+                cryptohash >= 0.7,
                 maccatcher >= 1.0 && < 2.2,
                 random >= 1.0.1 && < 1.1,
                 time >= 1.1 && < 1.5
