--- old/virthualenv.cabal	2012-09-05 22:50:47.809090845 -0700
+++ new/virthualenv.cabal	2012-09-05 22:49:55.521089609 -0700
@@ -103,10 +103,10 @@
                , filepath >= 1.1.0.3 && < 1.4
                , directory >= 1.0.1.0 && < 1.2
                , Cabal >= 1.8.0.6 && < 1.15
-               , mtl >= 1.1.0.2 && < 2.1
+               , mtl >= 1.1.0.2
                , bytestring >= 0.9.1.7 && < 0.10
                , file-embed >= 0.0.4.1 && < 0.1
-               , split >= 0.1.4 && < 0.2
+               , split >= 0.1.4
                , safe >= 0.3 && < 0.4
 
   Other-modules: Util.Cabal
