--- old/wl-pprint-text.cabal	2014-07-19 08:00:23.836465515 -0700
+++ new/wl-pprint-text.cabal	2014-10-01 09:22:08.542185341 -0700
@@ -19,4 +19,4 @@
   Exposed-modules:     Text.PrettyPrint.Leijen.Text,
                        Text.PrettyPrint.Leijen.Text.Monadic
   Build-depends:       base < 5,
-                       text >= 0.11.0.0 && < 1.2.0.0
+                       text >= 0.11.0.0 && < 1.3.0.0
