--- old/xml-enumerator.cabal	2012-01-20 06:47:15.000000000 -0800
+++ new/xml-enumerator.cabal	2012-01-20 09:54:58.246541244 -0800
@@ -34,17 +34,17 @@
 library
     build-depends:   base                      >= 4        && < 5
                    , enumerator                >= 0.4.5    && < 0.5
-                   , bytestring                >= 0.9      && < 0.10
+                   , bytestring                >= 0.9
                    , text                      >= 0.7      && < 0.12
-                   , containers                >= 0.2      && < 0.5
+                   , containers                >= 0.2
                    , xml-types                 >= 0.3      && < 0.4
                    , attoparsec-enumerator     >= 0.3
                    , attoparsec                >= 0.10
                    , blaze-builder             >= 0.2      && < 0.4
                    , blaze-builder-enumerator  >= 0.2      && < 0.3
-                   , transformers              >= 0.2      && < 0.3
-                   , failure                   >= 0.1      && < 0.2
-                   , data-default              >= 0.2      && < 0.4
+                   , transformers              >= 0.2
+                   , failure                   >= 0.1
+                   , data-default              >= 0.2
     exposed-modules: Text.XML.Stream.Parse
                      Text.XML.Stream.Render
                      Text.XML.Unresolved
@@ -59,11 +59,11 @@
     main-is: main.hs
     hs-source-dirs: test
     build-depends:          base                         >= 4               && < 5
-                          , containers                   >= 0.2             && < 0.5
+                          , containers                   >= 0.2
                           , text                                               < 1
-                          , transformers                 >= 0.2             && < 0.3
+                          , transformers                 >= 0.2
                           , enumerator                   >= 0.4.14          && < 0.5
-                          , bytestring                   >= 0.9             && < 0.10
+                          , bytestring                   >= 0.9
                           , xml-enumerator               >= 0.4             && < 0.5
                           , hspec                        >= 0.6.1           && < 0.10
                           , HUnit                        >= 1.2             && < 1.3
